---
title: "Solutions"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Solutions}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

Suggested answers to the workshop questions are below. You might have some different code e.g. to customise the volcano plot as you like. Feel free to comment on any of these solutions in the workshop website as described [here](https://github.com/stemangiola/ABACBS2020_tidytranscriptomics/blob/master/CONTRIBUTING.md).

```{r out.width = "70%", message=FALSE, warning=FALSE}
# load libraries
library(airway)

# tidyverse core packages
library(tibble)
library(dplyr)
library(tidyr)
library(readr)
library(stringr)
library(purrr)
library(ggplot2)

# tidyverse-friendly packages
library(plotly)
library(ggrepel)
library(tidyHeatmap)
library(tidybulk)
```

# Part 1 Bulk RNA-seq Core

## Airway dataset

```{r}
# Set up data
data("airway")

counts_tt <- 
    airway %>%
    tidybulk() %>%
    keep_abundant(factor_of_interest=dex)
```

### What fraction of variance is explained by PC3?

```{r}
counts_tt %>% 
  scale_abundance() %>%
  reduce_dimensions(method="PCA", .dims=3)
```


## Comparison of methods

### Which method detects the most differentially abundant transcripts, p value adjusted for multiple testing <  0.05 (FDR, adj.P.Val, padj)?

```{r}
# Set up data
de_all <- 
  
  counts_scal_PCA %>%
  
  # edgeR QLT
  test_differential_abundance(
    ~ dex + cell, 
    method = "edger_quasi_likelihood",
    prefix = "edgerQLT_"
  )  %>%
  
  # edgeR LRT
  test_differential_abundance(
    ~ dex + cell, 
    method = "edger_likelihood_ratio",
    prefix = "edgerLR_"
  )  %>%
  
  # limma-voom
  test_differential_abundance(
    ~ dex + cell, 
    method = "limma_voom",
    prefix = "voom_"
  ) %>%
  
  # DESeq2
  test_differential_abundance(
    ~ dex + cell,  
    method = "deseq2",
    prefix = "deseq2_"
  ) 
```

```{r out.width = "70%"}
de_all %>%
	
	# Subset transcript information
	pivot_transcript() %>%
	
	# Reshape for nesting
	pivot_longer(
		cols = -c(feature, .abundant, albut, symbol),
		names_sep = "_", 
		names_to = c("method", "statistic"), 
		values_to = "value"
	) %>%
	
	# Filter statistic
	filter(statistic %in% c("FDR", "adj.P.Val", "padj")) %>%
	filter(value < 0.05) %>%
	
	# Nesting
	count(method)
```

## Cell type composition

### What is the most abundant cell type overall in BRCA samples?

```{r eval=FALSE}
BRCA_cell_type_long %>% 
	group_by(cell_type) %>% 
	summarise(m = median(proportion)) %>% 
	dplyr::arrange(dplyr::desc(m))
```
