<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RNAseq-R-tidyverse • RNAseqRtidyverse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RNAseq-R-tidyverse">
<meta property="og:description" content="RNAseqRtidyverse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RNAseqRtidyverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/tidytranscriptomics.html">Material</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mblue9/RNAseq-R-tidyverse">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://hub.docker.com/repository/docker/mblue9/RNAseq-R-tidyverse">
    <span class="fa fa-docker"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tidytranscriptomics_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="tidytranscriptomics_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="tidytranscriptomics_files/anchor-sections-1.0/anchor-sections.js"></script><script src="tidytranscriptomics_files/htmlwidgets-1.5.2/htmlwidgets.js"></script><script src="tidytranscriptomics_files/plotly-binding-4.9.2.1/plotly.js"></script><script src="tidytranscriptomics_files/typedarray-0.1/typedarray.min.js"></script><script src="tidytranscriptomics_files/jquery-1.11.3/jquery.min.js"></script><link href="tidytranscriptomics_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="tidytranscriptomics_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><link href="tidytranscriptomics_files/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet">
<script src="tidytranscriptomics_files/plotly-main-1.52.2/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RNAseq-R-tidyverse</h1>
                        <h4 class="author">Maria Doyle, Peter MacCallum Cancer Centre<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">Stefano Mangiola, Walter and Eliza Hall Institute<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mblue9/RNAseq-R-tidyverse/blob/master/vignettes/tidytranscriptomics.Rmd"><code>vignettes/tidytranscriptomics.Rmd</code></a></small>
      <div class="hidden name"><code>tidytranscriptomics.Rmd</code></div>

    </div>

    
    
<style type="text/css">
# Formatting for polls
.poll {
background-color: #fff4d4;
}
.poll code {
    background-color: #fff4d4;
}
</style>
<p>This workshop will present how to perform analysis of RNA sequencing data following the tidy data paradigm <span class="citation">(Wickham and others 2014)</span>. The tidy data paradigm provides a standard way to organise data values within a dataset, where each variable is a column, each observation is a row, and data is manipulated using an easy-to-understand vocabulary. Most importantly, the data structure remains consistent across manipulation and analysis functions.</p>
<p>This can be achieved for RNA sequencing data with the <a href="https://stemangiola.github.io/tidybulk/">tidybulk</a>, <a href="https://stemangiola.github.io/tidyHeatmap">tidyHeatmap</a> <span class="citation">(Mangiola and Papenfuss 2020)</span> and <a href="https://www.tidyverse.org/">tidyverse</a> <span class="citation">(Wickham et al. 2019)</span> packages. The tidybulk package provides a tidy data structure and a modular framework for bulk transcriptional analyses. tidyHeatmap provides a tidy implementation of ComplexHeatmap. These packages are part of the tidytranscriptomics suite that introduces a tidy approach to RNA sequencing data representation and analysis</p>
<div id="acknowledgements" class="section level3">
<h3 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h3>
<p>This material was adapted from an R for RNA sequencing workshop first run <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">here</a>.</p>
<p><img src="../inst/vignettes/tidybulk_logo.png" width="100px"></p>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Measuring gene expression on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene expression in often complex systems. As sequencing costs have decreased, using RNA sequencing to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.</p>
<p>There are many steps involved in analysing an RNA sequencing dataset. Sequenced reads are aligned to a reference genome, then the number of reads mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on in R. While mapping and counting are important and necessary tasks, today we will be starting from the count data and showing how differential expression analysis can be performed in a friendly way using the Bioconductor package, tidybulk.</p>
<p>First, let’s load all the packages we will need to analyse the data.</p>
<p><em>Note: you should load the <em>tidybulk</em> library after the tidyverse core packages for best integration.</em></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load libraries</span>

<span class="co"># tidyverse core packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/">tibble</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org">readr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>

<span class="co"># tidyverse-friendly packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com">plotly</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggobi.github.io/ggally">GGally</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.r-project.org">tidyHeatmap</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">tidybulk</span><span class="op">)</span></code></pre></div>
<p>Plot settings. Set the colours and theme we will use for our plots.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use colourblind-friendly colours</span>
<span class="va">friendly_cols</span> <span class="op">&lt;-</span> <span class="fu">dittoSeq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dittoSeq/man/dittoColors.html">dittoColors</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Set theme</span>
<span class="va">custom_theme</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">friendly_cols</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">friendly_cols</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
        panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        axis.line <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
        panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,
        text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">10</span>, r <span class="op">=</span> <span class="fl">10</span>, b <span class="op">=</span> <span class="fl">10</span>, l <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">10</span>, r <span class="op">=</span> <span class="fl">10</span>, b <span class="op">=</span> <span class="fl">10</span>, l <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">30</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
      <span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div id="setting-up-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h3>
<p>Here we will perform RNA-Seq analysis using data from a breast cancer research study, from the paper by <a href="https://www.ncbi.nlm.nih.gov/pubmed/25730472">Fu et al. 2015</a>, GEO code GSE60450. This study examined gene expression in basal and luminal cells from mice at different stages of mammary gland development (virgin, pregnant and lactating). There are 2 samples per group and 6 groups, 12 samples in total.</p>
<p>We will be using the tidyverse pipe <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code>. This ‘pipes’ the output from the command on the left into the command on the right/below. Using the pipe is not essential but it reduces the amount of code we need to write when we have multiple steps (as we’ll see later). It also can make the steps clearer and easier to see. For more details on the pipe see <a href="https://r4ds.had.co.nz/pipes.html">here</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># import RNA-seq counts</span>
<span class="va">seqdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"https://ndownloader.figshare.com/files/5057929?private_link=1d788fd384d33e913a2a"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   EntrezGeneID = col_double(),</span>
<span class="co">#&gt;   Length = col_double(),</span>
<span class="co">#&gt;   `MCL1-DG_BC2CTUACXX_ACTTGA_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DH_BC2CTUACXX_CAGATC_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DI_BC2CTUACXX_ACAGTG_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DL_BC2CTUACXX_ATCACG_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LA_BC2CTUACXX_GATCAG_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LB_BC2CTUACXX_TGACCA_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LE_BC2CTUACXX_TAGCTT_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LF_BC2CTUACXX_CTTGTA_L001_R1` = col_double()</span>
<span class="co">#&gt; )</span>

<span class="co"># import sample information</span>
<span class="va">sampleinfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"https://ndownloader.figshare.com/files/5999832?private_link=1d788fd384d33e913a2a"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   FileName = col_character(),</span>
<span class="co">#&gt;   SampleName = col_character(),</span>
<span class="co">#&gt;   CellType = col_character(),</span>
<span class="co">#&gt;   Status = col_character()</span>
<span class="co">#&gt; )</span></code></pre></div>
<p>Take a look at the data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqdata</span>
<span class="co">#&gt; # A tibble: 27,179 x 14</span>
<span class="co">#&gt;    EntrezGeneID Length `MCL1-DG_BC2CTU… `MCL1-DH_BC2CTU… `MCL1-DI_BC2CTU…</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;</span>
<span class="co">#&gt;  1       497097   3634              438              300               65</span>
<span class="co">#&gt;  2    100503874   3259                1                0                1</span>
<span class="co">#&gt;  3    100038431   1634                0                0                0</span>
<span class="co">#&gt;  4        19888   9747                1                1                0</span>
<span class="co">#&gt;  5        20671   3130              106              182               82</span>
<span class="co">#&gt;  6        27395   4203              309              234              337</span>
<span class="co">#&gt;  7        18777   2433              652              515              948</span>
<span class="co">#&gt;  8    100503730    799                0                1                0</span>
<span class="co">#&gt;  9        21399   2847             1604             1495             1721</span>
<span class="co">#&gt; 10        58175   2241                4                2               14</span>
<span class="co">#&gt; # … with 27,169 more rows, and 9 more variables:</span>
<span class="co">#&gt; #   `MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-DL_BC2CTUACXX_ATCACG_L002_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LA_BC2CTUACXX_GATCAG_L001_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LB_BC2CTUACXX_TGACCA_L001_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LE_BC2CTUACXX_TAGCTT_L001_R1` &lt;dbl&gt;,</span>
<span class="co">#&gt; #   `MCL1-LF_BC2CTUACXX_CTTGTA_L001_R1` &lt;dbl&gt;</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleinfo</span>
<span class="co">#&gt; # A tibble: 12 x 4</span>
<span class="co">#&gt;    FileName                          SampleName CellType Status  </span>
<span class="co">#&gt;    &lt;chr&gt;                             &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;   </span>
<span class="co">#&gt;  1 MCL1.DG_BC2CTUACXX_ACTTGA_L002_R1 MCL1.DG    basal    virgin  </span>
<span class="co">#&gt;  2 MCL1.DH_BC2CTUACXX_CAGATC_L002_R1 MCL1.DH    basal    virgin  </span>
<span class="co">#&gt;  3 MCL1.DI_BC2CTUACXX_ACAGTG_L002_R1 MCL1.DI    basal    pregnant</span>
<span class="co">#&gt;  4 MCL1.DJ_BC2CTUACXX_CGATGT_L002_R1 MCL1.DJ    basal    pregnant</span>
<span class="co">#&gt;  5 MCL1.DK_BC2CTUACXX_TTAGGC_L002_R1 MCL1.DK    basal    lactate </span>
<span class="co">#&gt;  6 MCL1.DL_BC2CTUACXX_ATCACG_L002_R1 MCL1.DL    basal    lactate </span>
<span class="co">#&gt;  7 MCL1.LA_BC2CTUACXX_GATCAG_L001_R1 MCL1.LA    luminal  virgin  </span>
<span class="co">#&gt;  8 MCL1.LB_BC2CTUACXX_TGACCA_L001_R1 MCL1.LB    luminal  virgin  </span>
<span class="co">#&gt;  9 MCL1.LC_BC2CTUACXX_GCCAAT_L001_R1 MCL1.LC    luminal  pregnant</span>
<span class="co">#&gt; 10 MCL1.LD_BC2CTUACXX_GGCTAC_L001_R1 MCL1.LD    luminal  pregnant</span>
<span class="co">#&gt; 11 MCL1.LE_BC2CTUACXX_TAGCTT_L001_R1 MCL1.LE    luminal  lactate </span>
<span class="co">#&gt; 12 MCL1.LF_BC2CTUACXX_CTTGTA_L001_R1 MCL1.LF    luminal  lactate</span></code></pre></div>
<p>Shorten the sample names to the first 7 characters in the counts file columns.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">countdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename_with</a></span><span class="op">(</span><span class="va">seqdata</span>, <span class="op">~</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span><span class="op">(</span><span class="va">.x</span>, <span class="fl">1</span>, <span class="fl">7</span><span class="op">)</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"MCL"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Add gene symbols</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">countdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="va">countdata</span>, symbol <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Mm.eg.db/man/org.Mm.egBASE.html">org.Mm.eg.db</a></span>, 
                                          keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">EntrezGeneID</span><span class="op">)</span>, 
                                          keytype <span class="op">=</span> <span class="st">"ENTREZID"</span>, 
                                          column<span class="op">=</span><span class="st">"SYMBOL"</span>, 
                                          multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></code></pre></div>
<p>Convert the counts into long format (tidy format).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_long</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span><span class="va">countdata</span>, cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"MCL"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"sample"</span>, values_to <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span> 

<span class="co"># take a look</span>
<span class="va">counts_long</span>
<span class="co">#&gt; # A tibble: 326,148 x 5</span>
<span class="co">#&gt;    EntrezGeneID Length symbol sample  counts</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt;  1       497097   3634 Xkr4   MCL1-DG    438</span>
<span class="co">#&gt;  2       497097   3634 Xkr4   MCL1-DH    300</span>
<span class="co">#&gt;  3       497097   3634 Xkr4   MCL1-DI     65</span>
<span class="co">#&gt;  4       497097   3634 Xkr4   MCL1-DJ    237</span>
<span class="co">#&gt;  5       497097   3634 Xkr4   MCL1-DK    354</span>
<span class="co">#&gt;  6       497097   3634 Xkr4   MCL1-DL    287</span>
<span class="co">#&gt;  7       497097   3634 Xkr4   MCL1-LA      0</span>
<span class="co">#&gt;  8       497097   3634 Xkr4   MCL1-LB      0</span>
<span class="co">#&gt;  9       497097   3634 Xkr4   MCL1-LC      0</span>
<span class="co">#&gt; 10       497097   3634 Xkr4   MCL1-LD      0</span>
<span class="co">#&gt; # … with 326,138 more rows</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleinfo_formatted</span> <span class="op">&lt;-</span> <span class="va">sampleinfo</span> <span class="op">%&gt;%</span>
    
    <span class="co"># make column called Group by combining the CellType and Status columns</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html">unite</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="va">CellType</span><span class="op">:</span><span class="va">Status</span>, sep<span class="op">=</span><span class="st">"."</span>, remove<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
    <span class="co"># replace the . in the SampleName column with - so can join to counts</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>SampleName<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">SampleName</span>, <span class="st">"\\."</span>, <span class="st">"-"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Join the counts and sampleinfo</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">counts_long</span>, <span class="va">sampleinfo_formatted</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span> <span class="op">=</span> <span class="st">"SampleName"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># convert to tidybulk tibble</span>
<span class="va">counts_tt</span> <span class="op">&lt;-</span>
  <span class="va">counts_annot</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>EntrezGeneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">EntrezGeneID</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/tidybulk-methods.html">tidybulk</a></span><span class="op">(</span>.sample<span class="op">=</span><span class="va">sample</span>, .transcript<span class="op">=</span><span class="va">EntrezGeneID</span>, .abundance<span class="op">=</span><span class="va">counts</span><span class="op">)</span></code></pre></div>
<p>The <code>counts_airway</code> object contains information about genes and samples, the first column has the Ensembl gene identifier, the second column has the sample identifier and the third column has the gene transcription abundance. The abundance is the number of reads aligning to the gene in each experimental sample. The remaining columns include sample-wise information. The dex column tells us whether the samples are treated or untreated and the cell column tells us what cell line they are from. We can shorten the sample names. We can remove the SRR1039 prefix that’s present in all of them, as shorter names can fit better in some of the plots we will create. We can use <code><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate()</a></code> together with <code><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace()</a></code> to remove the SRR1039 string from the sample column. We can get the gene symbols for these Ensembl gene ids using the Bioconductor annotation package for human, <code>org.Hs.eg.db</code> and add them as a column using <code>mutate</code> again. With tidyverse, all above steps can be linked with the <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code>, as shown below. This has the benefits that * no temporary variables need to be created * less typing is required * the steps can be seen more clearly.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># setup data workflow</span>
<span class="va">counts_tt</span> <span class="op">&lt;-</span>
  <span class="va">airway</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/tidybulk-methods.html">tidybulk</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">sample</span>, <span class="st">"SRR1039"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va">org.Hs.eg.db</span>,
    keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">feature</span><span class="op">)</span>,
    keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,
    column <span class="op">=</span> <span class="st">"SYMBOL"</span>,
    multiVals <span class="op">=</span> <span class="st">"first"</span>
  <span class="op">)</span><span class="op">)</span>
<span class="co"># take a look</span>
<span class="va">counts_tt</span></code></pre></div>
<p>From this tidybulk tibble, we can perform differential expression analysis with the tidybulk package.</p>
</div>
</div>
<div id="filtering-lowly-transcribed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-lowly-transcribed-genes" class="anchor"></a>Filtering lowly transcribed genes</h2>
<p>Genes with very low counts across all libraries provide little evidence for differential expression and they can interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes should be filtered out prior to further analysis.</p>
<p>We can perform the filtering using tidybulk <code>keep_abundant</code> or <code>identify_abundant</code>. These functions can use the <em>edgeR</em> <code>filterByExpr</code> function described in <span class="citation">(Law et al. 2016)</span> to automatically identify the genes with adequate abundance for differential expression testing. By default, this will keep genes with ~10 counts in a minimum number of samples, the number of the samples in the smallest group. In this dataset the smallest group size is four (as we have four dex-treated samples versus four untreated). Alternatively, we could use <code>identify_abundant</code> to identify which genes are abundant or not (TRUE/FALSE), rather than just keeping the abundant ones.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filtering counts</span>
<span class="va">counts_filtered</span> <span class="op">&lt;-</span> <span class="va">counts_tt</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/keep_abundant-methods.html">keep_abundant</a></span><span class="op">(</span>factor_of_interest <span class="op">=</span> <span class="va">Group</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_filtered</span>
<span class="co">#&gt; # A tibble: 191,628 x 10</span>
<span class="co">#&gt;    EntrezGeneID Length symbol sample counts FileName Group CellType Status</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; </span>
<span class="co">#&gt;  1 497097         3634 Xkr4   MCL1-…    438 MCL1.DG… basa… basal    virgin</span>
<span class="co">#&gt;  2 497097         3634 Xkr4   MCL1-…    300 MCL1.DH… basa… basal    virgin</span>
<span class="co">#&gt;  3 497097         3634 Xkr4   MCL1-…     65 MCL1.DI… basa… basal    pregn…</span>
<span class="co">#&gt;  4 497097         3634 Xkr4   MCL1-…    237 MCL1.DJ… basa… basal    pregn…</span>
<span class="co">#&gt;  5 497097         3634 Xkr4   MCL1-…    354 MCL1.DK… basa… basal    lacta…</span>
<span class="co">#&gt;  6 497097         3634 Xkr4   MCL1-…    287 MCL1.DL… basa… basal    lacta…</span>
<span class="co">#&gt;  7 497097         3634 Xkr4   MCL1-…      0 MCL1.LA… lumi… luminal  virgin</span>
<span class="co">#&gt;  8 497097         3634 Xkr4   MCL1-…      0 MCL1.LB… lumi… luminal  virgin</span>
<span class="co">#&gt;  9 497097         3634 Xkr4   MCL1-…      0 MCL1.LC… lumi… luminal  pregn…</span>
<span class="co">#&gt; 10 497097         3634 Xkr4   MCL1-…      0 MCL1.LD… lumi… luminal  pregn…</span>
<span class="co">#&gt; # … with 191,618 more rows, and 1 more variable: .abundant &lt;lgl&gt;</span></code></pre></div>
<p>After running <code>keep_abundant</code> we have a column called <code>.abundant</code> containing TRUE (<code>identify_abundant</code> would have TRUE/FALSE).</p>
</div>
<div id="scaling-counts-to-normalise" class="section level2">
<h2 class="hasAnchor">
<a href="#scaling-counts-to-normalise" class="anchor"></a>Scaling counts to normalise</h2>
<p>Scaling of counts, normalisation, is performed to eliminate uninteresting differences between samples due to sequencing depth or composition. A more detailed explanation can be found <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>. In the tidybulk package the function <code>scale_abundance</code> generates scaled counts, with scaling factors calculated on abundant (filtered) transcripts and applied to all transcripts. We can choose from different normalisation methods. Here we will use the default, edgeR’s trimmed mean of M values (TMM), <span class="citation">(Robinson and Oshlack 2010)</span>. TMM normalisation (and most scaling normalisation methods) scale relative to one sample.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Scaling counts</span>
<span class="va">counts_scaled</span> <span class="op">&lt;-</span> <span class="va">counts_filtered</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/scale_abundance-methods.html">scale_abundance</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_scaled</span>
<span class="co">#&gt; # A tibble: 191,628 x 13</span>
<span class="co">#&gt;    EntrezGeneID Length symbol sample counts FileName Group CellType Status</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; </span>
<span class="co">#&gt;  1 497097         3634 Xkr4   MCL1-…    438 MCL1.DG… basa… basal    virgin</span>
<span class="co">#&gt;  2 497097         3634 Xkr4   MCL1-…    300 MCL1.DH… basa… basal    virgin</span>
<span class="co">#&gt;  3 497097         3634 Xkr4   MCL1-…     65 MCL1.DI… basa… basal    pregn…</span>
<span class="co">#&gt;  4 497097         3634 Xkr4   MCL1-…    237 MCL1.DJ… basa… basal    pregn…</span>
<span class="co">#&gt;  5 497097         3634 Xkr4   MCL1-…    354 MCL1.DK… basa… basal    lacta…</span>
<span class="co">#&gt;  6 497097         3634 Xkr4   MCL1-…    287 MCL1.DL… basa… basal    lacta…</span>
<span class="co">#&gt;  7 497097         3634 Xkr4   MCL1-…      0 MCL1.LA… lumi… luminal  virgin</span>
<span class="co">#&gt;  8 497097         3634 Xkr4   MCL1-…      0 MCL1.LB… lumi… luminal  virgin</span>
<span class="co">#&gt;  9 497097         3634 Xkr4   MCL1-…      0 MCL1.LC… lumi… luminal  pregn…</span>
<span class="co">#&gt; 10 497097         3634 Xkr4   MCL1-…      0 MCL1.LD… lumi… luminal  pregn…</span>
<span class="co">#&gt; # … with 191,618 more rows, and 4 more variables: .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;,</span>
<span class="co">#&gt; #   multiplier &lt;dbl&gt;, counts_scaled &lt;dbl&gt;</span></code></pre></div>
<p>After we run <code>scale_abundance</code> we should see some columns have been added at the end. The <code>counts_scaled</code> column contains the scaled counts.</p>
<p>We can visualise the difference of abundance distributions before and after scaling. As tidybulk output is compatible with tidyverse, we can simply pipe it into standard tidyverse functions such as <code>filter</code>, <code>pivot_longer</code> and <code>ggplot</code>. We can also take advantage of ggplot’s <code>facet_wrap</code> to easily create multiple plots.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_scaled</span> <span class="op">%&gt;%</span>

  <span class="co"># Reshaping        </span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"counts"</span>, <span class="st">"counts_scaled"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"source"</span>, values_to <span class="op">=</span> <span class="st">"abundance"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
  <span class="co"># Plotting</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="va">abundance</span> <span class="op">+</span> <span class="fl">1</span>, fill <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">abundance</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">source</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-18-1.png" width="70%"></p>
<p>In this dataset the distributions of the counts are not very different to each other before scaling but scaling does make the distributions more similar. If we saw a sample with a very different distribution we may need to investigate it.</p>
</div>
<div id="exploratory-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-analyses" class="anchor"></a>Exploratory analyses</h2>
<div id="dimensionality-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#dimensionality-reduction" class="anchor"></a>Dimensionality reduction</h3>
<p>By far, one of the most important plots we make when we analyse RNA sequencing data are principal-component analysis (PCA) or multi-dimensional scaling (MDS) plots. We reduce the dimensions of the data to identify the greatest sources of variation in the data. A principal components analysis is an example of an unsupervised analysis, where we don’t need to specify the groups. If your experiment is well controlled and has worked well, what we hope to see is that the greatest sources of variation in the data are the treatments/groups we are interested in. It is also an incredibly useful tool for quality control and checking for outliers. We can use the <code>reduce_dimensions</code> function to calculate the dimensions.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get principal components</span>
<span class="va">counts_scal_MDS</span> <span class="op">&lt;-</span>
  <span class="va">counts_scaled</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/reduce_dimensions-methods.html">reduce_dimensions</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"MDS"</span>, scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; tidybulk says: to access the raw results do `attr(..., "internals")$MDS`</span></code></pre></div>
<p>This joins the result to the counts object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Take a look</span>
<span class="va">counts_scal_MDS</span>
<span class="co">#&gt; # A tibble: 191,628 x 15</span>
<span class="co">#&gt;    EntrezGeneID Length symbol sample counts FileName Group CellType Status</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; </span>
<span class="co">#&gt;  1 497097         3634 Xkr4   MCL1-…    438 MCL1.DG… basa… basal    virgin</span>
<span class="co">#&gt;  2 497097         3634 Xkr4   MCL1-…    300 MCL1.DH… basa… basal    virgin</span>
<span class="co">#&gt;  3 497097         3634 Xkr4   MCL1-…     65 MCL1.DI… basa… basal    pregn…</span>
<span class="co">#&gt;  4 497097         3634 Xkr4   MCL1-…    237 MCL1.DJ… basa… basal    pregn…</span>
<span class="co">#&gt;  5 497097         3634 Xkr4   MCL1-…    354 MCL1.DK… basa… basal    lacta…</span>
<span class="co">#&gt;  6 497097         3634 Xkr4   MCL1-…    287 MCL1.DL… basa… basal    lacta…</span>
<span class="co">#&gt;  7 497097         3634 Xkr4   MCL1-…      0 MCL1.LA… lumi… luminal  virgin</span>
<span class="co">#&gt;  8 497097         3634 Xkr4   MCL1-…      0 MCL1.LB… lumi… luminal  virgin</span>
<span class="co">#&gt;  9 497097         3634 Xkr4   MCL1-…      0 MCL1.LC… lumi… luminal  pregn…</span>
<span class="co">#&gt; 10 497097         3634 Xkr4   MCL1-…      0 MCL1.LD… lumi… luminal  pregn…</span>
<span class="co">#&gt; # … with 191,618 more rows, and 6 more variables: .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;,</span>
<span class="co">#&gt; #   multiplier &lt;dbl&gt;, counts_scaled &lt;dbl&gt;, Dim1 &lt;dbl&gt;, Dim2 &lt;dbl&gt;</span></code></pre></div>
<p>For plotting, we can select just the sample-wise information with <code>pivot_sample</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take a look</span>
<span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_sample-methods.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 x 10</span>
<span class="co">#&gt;    sample FileName Group CellType Status .abundant   TMM multiplier  Dim1   Dim2</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;  &lt;lgl&gt;     &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1 MCL1-… MCL1.DG… basa… basal    virgin TRUE      1.25       0.854 -2.00  0.838</span>
<span class="co">#&gt;  2 MCL1-… MCL1.DH… basa… basal    virgin TRUE      1.23       0.926 -2.03  0.740</span>
<span class="co">#&gt;  3 MCL1-… MCL1.DI… basa… basal    pregn… TRUE      1.13       0.910 -2.08 -0.315</span>
<span class="co">#&gt;  4 MCL1-… MCL1.DJ… basa… basal    pregn… TRUE      1.08       1.01  -2.26 -0.347</span>
<span class="co">#&gt;  5 MCL1-… MCL1.DK… basa… basal    lacta… TRUE      1.03       1.12  -2.41 -0.755</span>
<span class="co">#&gt;  6 MCL1-… MCL1.DL… basa… basal    lacta… TRUE      1.08       1.15  -2.39 -0.758</span>
<span class="co">#&gt;  7 MCL1-… MCL1.LA… lumi… luminal  virgin TRUE      1.41       0.859  1.80  2.12 </span>
<span class="co">#&gt;  8 MCL1-… MCL1.LB… lumi… luminal  virgin TRUE      1.44       0.794  1.65  2.05 </span>
<span class="co">#&gt;  9 MCL1-… MCL1.LC… lumi… luminal  pregn… TRUE      0.984      1.13   2.23 -0.205</span>
<span class="co">#&gt; 10 MCL1-… MCL1.LD… lumi… luminal  pregn… TRUE      0.902      1.25   2.30 -0.325</span>
<span class="co">#&gt; 11 MCL1-… MCL1.LE… lumi… luminal  lacta… TRUE      0.519      1.93   2.65 -1.48 </span>
<span class="co">#&gt; 12 MCL1-… MCL1.LF… lumi… luminal  lacta… TRUE      0.522      1.92   2.53 -1.56</span></code></pre></div>
<p>We can now plot the reduced dimensions.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MDS plot</span>
<span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_sample-methods.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, colour <span class="op">=</span> <span class="va">CellType</span>, shape <span class="op">=</span> <span class="va">Status</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">sample</span><span class="op">)</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-22-1.png" width="70%"></p>
<p>The samples separate by treatment on PC1 which is what we hope to see. PC2 separates the N080611 cell line from the other samples, indicating a greater difference between that cell line and the others.</p>
</div>
<div id="hierarchical-clustering-with-heatmaps" class="section level3">
<h3 class="hasAnchor">
<a href="#hierarchical-clustering-with-heatmaps" class="anchor"></a>Hierarchical clustering with heatmaps</h3>
<p>An alternative to principal component analysis for examining relationships between samples is using hierarchical clustering. Heatmaps are a nice visualisation to examine hierarchical clustering of your samples. tidybulk has a simple function we can use, <code>keep_variable</code>, to extract the most variable genes which we can then plot with tidyHeatmap.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>

  <span class="co"># extract 500 most variable genes</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/keep_variable-methods.html">keep_variable</a></span><span class="op">(</span>.abundance <span class="op">=</span> <span class="va">counts_scaled</span>, top <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># create heatmap</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/heatmap-methods.html">heatmap</a></span><span class="op">(</span>
    .column <span class="op">=</span> <span class="va">sample</span>,
    .row <span class="op">=</span> <span class="va">EntrezGeneID</span>,
    .value <span class="op">=</span> <span class="va">counts_scaled</span>,
    transform <span class="op">=</span> <span class="va">log1p</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_tile-methods.html">add_tile</a></span><span class="op">(</span><span class="va">CellType</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_tile-methods.html">add_tile</a></span><span class="op">(</span><span class="va">Status</span><span class="op">)</span>
<span class="co">#&gt; Getting the 500 most variable genes</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-23-1.png" width="70%"></p>
<p>In the heatmap we can see the samples cluster into two groups, treated and untreated, for three of the cell lines, and the cell line (N080611) again is further away from the others.</p>
<p>Tidybulk enables a simplified way of generating a clustered heatmap of variable genes. Compare the code below for tidybulk versus a base R method.</p>
<p><strong>base R using edgeR</strong></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example code, no need to run</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/SE2DGEList.html">SE2DGEList</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>
<span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dgList</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">dex</span><span class="op">)</span>
<span class="va">keep.exprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">dgList</span>, group <span class="op">=</span> <span class="va">group</span><span class="op">)</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="va">dgList</span><span class="op">[</span><span class="va">keep.exprs</span>, , keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">dgList</span><span class="op">)</span>
<span class="va">logcounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">dgList</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">var_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">logcounts</span>, <span class="fl">1</span>, <span class="va">var</span><span class="op">)</span>
<span class="va">select_var</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">var_genes</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span><span class="op">]</span>
<span class="va">highly_variable_lcpm</span> <span class="op">&lt;-</span> <span class="va">logcounts</span><span class="op">[</span><span class="va">select_var</span>, <span class="op">]</span>
<span class="va">colours</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#440154FF"</span>, <span class="st">"#21908CFF"</span>, <span class="st">"#fefada"</span><span class="op">)</span>
<span class="va">col.group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"grey"</span><span class="op">)</span><span class="op">[</span><span class="va">group</span><span class="op">]</span>
<span class="fu">gplots</span><span class="fu">::</span><span class="fu">heatmap.2</span><span class="op">(</span><span class="va">highly_variable_lcpm</span>, col <span class="op">=</span> <span class="va">colours</span>, trace <span class="op">=</span> <span class="st">"none"</span>, ColSideColors <span class="op">=</span> <span class="va">col.group</span>, scale <span class="op">=</span> <span class="st">"row"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="differential-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential expression</h2>
<p><em>tidybulk</em> integrates several popular methods for differential transcript abundance testing: the edgeR quasi-likelihood <span class="citation">(Chen, Lun, and Smyth 2016)</span> (tidybulk default method), edgeR likelihood ratio <span class="citation">(McCarthy, Chen, and Smyth 2012)</span>, limma-voom <span class="citation">(Law et al. 2014)</span> and DESeq2 <span class="citation">(Love, Huber, and Anders 2014)</span>. A common question researchers have is which method to choose. With tidybulk we can easily run multiple methods and compare.</p>
<p>We give <code>test_differential_abundance</code> our tidybulk counts object and a formula, specifying the column that contains our groups to be compared. If all our samples were from the same cell line, and there were no additional factors contributing variance such as batch differences, we could use the formula <code>~ dex</code>. However, each treated and untreated sample is from a different cell line so we add the cell line as an additional factor <code>~ dex + cell</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_all</span> <span class="op">&lt;-</span>

  <span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>

  <span class="co"># edgeR QLT</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span>
    <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
    method <span class="op">=</span> <span class="st">"edgeR_quasi_likelihood"</span>,
    prefix <span class="op">=</span> <span class="st">"edgerQLT_"</span>,
    .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
    omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># edgeR LRT</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span>
    <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
    method <span class="op">=</span> <span class="st">"edgeR_likelihood_ratio"</span>,
    prefix <span class="op">=</span> <span class="st">"edgerLR_"</span>,
    .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
    omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># limma-voom</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span>
    <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
    method <span class="op">=</span> <span class="st">"limma_voom"</span>,
    prefix <span class="op">=</span> <span class="st">"voom_"</span>,
    .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
    omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># DESeq2</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span>
    <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
    method <span class="op">=</span> <span class="st">"deseq2"</span>,
    prefix <span class="op">=</span> <span class="st">"deseq2_"</span>,
    .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"basal.lactate"</span>, <span class="st">"basal.pregnant"</span><span class="op">)</span><span class="op">)</span>,
    omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span>

<span class="co"># take a look</span>

<span class="va">de_all</span>
<span class="co">#&gt; # A tibble: 191,628 x 37</span>
<span class="co">#&gt;    EntrezGeneID Length symbol sample counts FileName Group CellType Status</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; </span>
<span class="co">#&gt;  1 497097         3634 Xkr4   MCL1-…    438 MCL1.DG… basa… basal    virgin</span>
<span class="co">#&gt;  2 497097         3634 Xkr4   MCL1-…    300 MCL1.DH… basa… basal    virgin</span>
<span class="co">#&gt;  3 497097         3634 Xkr4   MCL1-…     65 MCL1.DI… basa… basal    pregn…</span>
<span class="co">#&gt;  4 497097         3634 Xkr4   MCL1-…    237 MCL1.DJ… basa… basal    pregn…</span>
<span class="co">#&gt;  5 497097         3634 Xkr4   MCL1-…    354 MCL1.DK… basa… basal    lacta…</span>
<span class="co">#&gt;  6 497097         3634 Xkr4   MCL1-…    287 MCL1.DL… basa… basal    lacta…</span>
<span class="co">#&gt;  7 497097         3634 Xkr4   MCL1-…      0 MCL1.LA… lumi… luminal  virgin</span>
<span class="co">#&gt;  8 497097         3634 Xkr4   MCL1-…      0 MCL1.LB… lumi… luminal  virgin</span>
<span class="co">#&gt;  9 497097         3634 Xkr4   MCL1-…      0 MCL1.LC… lumi… luminal  pregn…</span>
<span class="co">#&gt; 10 497097         3634 Xkr4   MCL1-…      0 MCL1.LD… lumi… luminal  pregn…</span>
<span class="co">#&gt; # … with 191,618 more rows, and 28 more variables: .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;,</span>
<span class="co">#&gt; #   multiplier &lt;dbl&gt;, counts_scaled &lt;dbl&gt;, Dim1 &lt;dbl&gt;, Dim2 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   edgerQLT_logFC &lt;dbl&gt;, edgerQLT_logCPM &lt;dbl&gt;, edgerQLT_F &lt;dbl&gt;,</span>
<span class="co">#&gt; #   edgerQLT_PValue &lt;dbl&gt;, edgerQLT_FDR &lt;dbl&gt;, edgerLR_logFC &lt;dbl&gt;,</span>
<span class="co">#&gt; #   edgerLR_logCPM &lt;dbl&gt;, edgerLR_LR &lt;dbl&gt;, edgerLR_PValue &lt;dbl&gt;,</span>
<span class="co">#&gt; #   edgerLR_FDR &lt;dbl&gt;, voom_logFC &lt;dbl&gt;, voom_AveExpr &lt;dbl&gt;, voom_t &lt;dbl&gt;,</span>
<span class="co">#&gt; #   voom_P.Value &lt;dbl&gt;, voom_adj.P.Val &lt;dbl&gt;, voom_B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   deseq2_baseMean &lt;dbl&gt;, deseq2_log2FoldChange &lt;dbl&gt;, deseq2_lfcSE &lt;dbl&gt;,</span>
<span class="co">#&gt; #   deseq2_stat &lt;dbl&gt;, deseq2_pvalue &lt;dbl&gt;, deseq2_padj &lt;dbl&gt;</span></code></pre></div>
<p>This outputs the columns from each method such as log-fold change (logFC), false-discovery rate (FDR) and probability value (p-value). logFC is log2(treated/untreated).</p>
<div id="comparison-of-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#comparison-of-methods" class="anchor"></a>Comparison of methods</h3>
<p>We can visually compare the significance for all methods. We will notice that there is some difference between the methods.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_all</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">edgerQLT_PValue</span>, <span class="va">edgerLR_PValue</span>, <span class="va">voom_P.Value</span>, <span class="va">deseq2_pvalue</span>, <span class="va">EntrezGeneID</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggobi.github.io/ggally/reference/ggpairs.html">ggpairs</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-26-1.png" width="700"></p>
<p>In order to decide which genes are differentially expressed, we usually take a cut-off of 0.05 on the FDR (or adjusted P value), NOT the raw p-value. This is because we are testing many genes (multiple testing), and the chances of finding differentially expressed genes is very high when you do that many tests. Hence we need to control the false discovery rate, which is the adjusted p-value column in the results table. What this means is that if 100 genes are significant at a 5% false discovery rate, we are willing to accept that 5 will be false positives.</p>
<p>We can count how many differentially expressed genes there are, for example with the edgeR QLT method. We’ll filter on FDR 0.05.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_all</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">edgerQLT_FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>num_de <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html">n_distinct</a></span><span class="op">(</span><span class="va">EntrezGeneID</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 1</span>
<span class="co">#&gt;   num_de</span>
<span class="co">#&gt;    &lt;int&gt;</span>
<span class="co">#&gt; 1   5287</span></code></pre></div>
<pre class="poll_7 poll"><code>Which method detects the most differentially abundant transcripts, p value adjusted for multiple testing &lt;  0.05 (FDR, adj.P.Val, padj)?</code></pre>
<p><em>Note: Some of the methods produce columns with different names for similar outputs. If you wish to make these consistent you can do that with tidyverse <code>rename</code>. For example, to rename the p value adjusted columns you could run below.</em></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_all</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>deseq2_FDR <span class="op">=</span> <span class="va">deseq2_padj</span>, voom_FDR <span class="op">=</span> <span class="va">voom_adj.P.Val</span><span class="op">)</span></code></pre></div>
<p>We can select some of the transcripts that look different between methods in the plot using the <a href="https://github.com/stemangiola/tidygate"><em>tidygate</em></a> package. We can then investigate these transcripts, for example, by visualising their counts.</p>
<p>With tidygate, we can interactively draw gates to select points we want using <code>gate</code>. We specify which columns we want to plot in the scatterplot, and how many gates we want to draw. We can also specify the opacity if we want to make it easier to see overlapping points.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_gate</span> <span class="op">&lt;-</span>
  <span class="va">de_all</span> <span class="op">%&gt;%</span>

  <span class="fu">tidygate</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidygate/man/gate-methods.html">gate</a></span><span class="op">(</span>
    <span class="va">EntrezGeneID</span>,
    <span class="va">edgerQLT_PValue</span>,
    <span class="va">deseq2_pvalue</span>,
    opacity <span class="op">=</span> <span class="fl">0.3</span>,
    how_many_gates <span class="op">=</span> <span class="fl">2</span>
  <span class="op">)</span></code></pre></div>
<p>We then click to draw gates around the points we want, for example as shown in the screenshot below.</p>
<p><img src="../inst/vignettes/comparison_different_DE_methods_gates.png" width="322"></p>
<p>That will add a column called gate, specifying which gate the points (transcripts) are in.</p>
<pre><code>#&gt; Warning: `gate()` is deprecated as of tidygate 0.3.0.
#&gt; Please use `gate_chr()` instead.
#&gt; This warning is displayed once every 8 hours.
#&gt; Call `lifecycle::last_warnings()` to see where this warning was generated.
#&gt; # A tibble: 191,628 x 38
#&gt;    EntrezGeneID Length symbol sample counts FileName Group CellType Status
#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; 
#&gt;  1 497097         3634 Xkr4   MCL1-…    438 MCL1.DG… basa… basal    virgin
#&gt;  2 497097         3634 Xkr4   MCL1-…    300 MCL1.DH… basa… basal    virgin
#&gt;  3 497097         3634 Xkr4   MCL1-…     65 MCL1.DI… basa… basal    pregn…
#&gt;  4 497097         3634 Xkr4   MCL1-…    237 MCL1.DJ… basa… basal    pregn…
#&gt;  5 497097         3634 Xkr4   MCL1-…    354 MCL1.DK… basa… basal    lacta…
#&gt;  6 497097         3634 Xkr4   MCL1-…    287 MCL1.DL… basa… basal    lacta…
#&gt;  7 497097         3634 Xkr4   MCL1-…      0 MCL1.LA… lumi… luminal  virgin
#&gt;  8 497097         3634 Xkr4   MCL1-…      0 MCL1.LB… lumi… luminal  virgin
#&gt;  9 497097         3634 Xkr4   MCL1-…      0 MCL1.LC… lumi… luminal  pregn…
#&gt; 10 497097         3634 Xkr4   MCL1-…      0 MCL1.LD… lumi… luminal  pregn…
#&gt; # … with 191,618 more rows, and 29 more variables: .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;,
#&gt; #   multiplier &lt;dbl&gt;, counts_scaled &lt;dbl&gt;, Dim1 &lt;dbl&gt;, Dim2 &lt;dbl&gt;,
#&gt; #   edgerQLT_logFC &lt;dbl&gt;, edgerQLT_logCPM &lt;dbl&gt;, edgerQLT_F &lt;dbl&gt;,
#&gt; #   edgerQLT_PValue &lt;dbl&gt;, edgerQLT_FDR &lt;dbl&gt;, edgerLR_logFC &lt;dbl&gt;,
#&gt; #   edgerLR_logCPM &lt;dbl&gt;, edgerLR_LR &lt;dbl&gt;, edgerLR_PValue &lt;dbl&gt;,
#&gt; #   edgerLR_FDR &lt;dbl&gt;, voom_logFC &lt;dbl&gt;, voom_AveExpr &lt;dbl&gt;, voom_t &lt;dbl&gt;,
#&gt; #   voom_P.Value &lt;dbl&gt;, voom_adj.P.Val &lt;dbl&gt;, voom_B &lt;dbl&gt;,
#&gt; #   deseq2_baseMean &lt;dbl&gt;, deseq2_log2FoldChange &lt;dbl&gt;, deseq2_lfcSE &lt;dbl&gt;,
#&gt; #   deseq2_stat &lt;dbl&gt;, deseq2_pvalue &lt;dbl&gt;, deseq2_padj &lt;dbl&gt;, gate &lt;chr&gt;</code></pre>
<p>We can check how many transcripts we’ve got in each gate.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_gate</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">gate</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 2</span>
<span class="co">#&gt;   gate      n</span>
<span class="co">#&gt;   &lt;chr&gt; &lt;int&gt;</span>
<span class="co">#&gt; 1 0     15969</span></code></pre></div>
<p>We can now select the transcripts from our two gates i.e. more significant in edgeR (gate 1) and more significant in DESeq2 (gate 2) and visualise the counts for each sample in the treated and untreated groups.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_gate</span> <span class="op">%&gt;%</span>

  <span class="co"># Filter for transcripts within the gates</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">gate</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Rename for clarity</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>gate <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span>
    <span class="va">gate</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">~</span> <span class="st">"more in edgeR"</span>,
    <span class="va">gate</span> <span class="op">==</span> <span class="fl">2</span> <span class="op">~</span> <span class="st">"more in DESeq2"</span>,
    <span class="cn">TRUE</span> <span class="op">~</span> <span class="va">gate</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Order the plots for the transcripts</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>EntrezGeneID <span class="op">=</span> <span class="fu">forcats</span><span class="fu">::</span><span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_reorder.html">fct_reorder</a></span><span class="op">(</span><span class="va">EntrezGeneID</span>, <span class="va">edgerQLT_PValue</span>, <span class="va">min</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Plot</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">dex</span>, <span class="va">counts_scaled</span>, color <span class="op">=</span> <span class="va">gate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">EntrezGeneID</span>, scale <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span></code></pre></div>
<p>We could also check the log fold changes for these genes and see, for example, that DESeq2 produces a more conservative logFC statistic for the gene <code>ENSG00000104725</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_gate</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">EntrezGeneID</span> <span class="op">==</span> <span class="st">"ENSG00000104725"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">edgerQLT_logFC</span>, <span class="va">deseq2_log2FoldChange</span><span class="op">)</span></code></pre></div>
</div>
<div id="single-method" class="section level3">
<h3 class="hasAnchor">
<a href="#single-method" class="anchor"></a>Single method</h3>
<p>If we just wanted to run one differential testing method we could do that. The default method is edgeR quasi-likelihood.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">&lt;-</span> <span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
                                .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
                                omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: All testing methods use raw counts, irrespective of if scale_abundance </span>
<span class="co">#&gt; or adjust_abundance have been calculated. Therefore, it is essential to add covariates </span>
<span class="co">#&gt; such as batch effects (if applicable) in the formula.</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: The design column names are "Groupbasal.lactate, Groupbasal.pregnant, Groupbasal.virgin, Groupluminal.lactate, Groupluminal.pregnant, Groupluminal.virgin"</span>
<span class="co">#&gt; tidybulk says: to access the raw results (fitted GLM) do `attr(..., "internals")$edgeR`</span></code></pre></div>
<p>Tidybulk enables a simplified way of performing an RNA sequencing differential expression analysis (with the benefit of smoothly integrating with ggplot2 and other tidyverse packages). Compare the code for a tidybulk edgeR analysis versus standard edgeR below.</p>
<p><strong>standard edgeR</strong></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example code, no need to run</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/SE2DGEList.html">SE2DGEList</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>
<span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dgList</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">dex</span><span class="op">)</span>
<span class="va">keep.exprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">dgList</span>, group <span class="op">=</span> <span class="va">group</span><span class="op">)</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="va">dgList</span><span class="op">[</span><span class="va">keep.exprs</span>, , keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">dgList</span><span class="op">)</span>
<span class="va">cell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dgList</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">cell</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">group</span> <span class="op">+</span> <span class="va">cell</span><span class="op">)</span>
<span class="va">dgList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">dgList</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFit</a></span><span class="op">(</span><span class="va">dgList</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">qlf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="plots-after-testing-for-differentially-expressed" class="section level2">
<h2 class="hasAnchor">
<a href="#plots-after-testing-for-differentially-expressed" class="anchor"></a>Plots after testing for differentially expressed</h2>
<p>We’ll extract the symbols for a few top genes (by P value) to use in some of the plots we will make.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">topgenes_symbols</span> <span class="op">&lt;-</span>
  <span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">PValue</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></code></pre></div>
<div id="volcano-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h3>
<p>Volcano plots are a useful genome-wide plot for checking that the analysis looks good. Volcano plots enable us to visualise the significance of change (p-value) versus the fold change (logFC). Highly significant genes are towards the top of the plot. We can also colour significant genes (e.g. genes with false-discovery rate &lt; 0.05)</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># volcano plot, minimal</span>
<span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="va">PValue</span>, colour <span class="op">=</span> <span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="st">"log10_reverse"</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-39-1.png" width="70%"></p>
<p>A more informative plot, integrating some of the packages in tidyverse.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Subset data</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>significant <span class="op">=</span> <span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">%in%</span> <span class="va">topgenes_symbols</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Plot</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="va">PValue</span>, label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">significant</span>, size <span class="op">=</span> <span class="va">significant</span>, alpha <span class="op">=</span> <span class="va">significant</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>

  <span class="co"># Custom scales</span>
  <span class="va">custom_theme</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="st">"log10_reverse"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"#e11f28"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_discrete</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'colour' is already present. Adding another scale for 'colour',</span>
<span class="co">#&gt; which will replace the existing scale.</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-40-1.png" width="70%"></p>
</div>
<div id="stripcharts" class="section level3">
<h3 class="hasAnchor">
<a href="#stripcharts" class="anchor"></a>Stripcharts</h3>
<p>Before following up on the differentially expressed genes with further lab work, it is also recommended to have a look at the expression levels of the individual samples for the genes of interest. We can use stripcharts to do this. These will help show if expression is consistent amongst replicates in the groups.</p>
<p>With stripcharts we can see if replicates tend to group together and how the expression compares to the other groups. We’ll also add a box plot to show the distribution. Tidyverse faceting makes it easy to create a plot for each gene.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strip_chart</span> <span class="op">&lt;-</span>
  <span class="va">counts_scaled</span> <span class="op">%&gt;%</span>

  <span class="co"># extract counts for top differentially expressed genes</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">%in%</span> <span class="va">topgenes_symbols</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># make faceted stripchart</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Group</span>, y <span class="op">=</span> <span class="va">counts_scaled</span> <span class="op">+</span> <span class="fl">1</span>, fill <span class="op">=</span> <span class="va">Group</span>, label <span class="op">=</span> <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">symbol</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span>

<span class="va">strip_chart</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-41-1.png" width="70%"></p>
</div>
</div>
<div id="interactive-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#interactive-plots" class="anchor"></a>Interactive Plots</h2>
<p>A really nice feature of using tidyverse and ggplot2 is that we can make interactive plots quite easily using the plotly package. This can be very useful for exploring what genes or samples are in the plots. We can make interactive plots directly from our ggplot2 object (strip_chart). Having <code>label</code> in the <code>aes</code> is useful to visualise the identifier of the data point (here the sample id) or other variables when we hover over the plot.</p>
<p>We can also specify which parameters from the <code>aes</code> we want to show up when we hover over the plot with <code>tooltip</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strip_chart</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/plotly/man/ggplotly.html">ggplotly</a></span><span class="op">(</span>tooltip <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"label"</span>, <span class="st">"y"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-0031ed1855e0d14d003e" style="width:70%;height:432.632880098888px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-0031ed1855e0d14d003e">{"x":{"data":[{"x":[1,1],"y":[3.91467864441465,3.82834807703469],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,1)","legendgroup":"(basal.lactate,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[1,1],"y":[1.81190883068896,1.71089291656358],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,2)","legendgroup":"(basal.lactate,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[1,1],"y":[3.20398045748516,3.14415157661807],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,3)","legendgroup":"(basal.lactate,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[1,1],"y":[2.94520040544604,2.99736252191693],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,4)","legendgroup":"(basal.lactate,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[1,1],"y":[3.50578645073982,3.42791293458927],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,5)","legendgroup":"(basal.lactate,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[1,1],"y":[3.39404893736358,3.21903646036174],"hoverinfo":"y","type":"box","fillcolor":"rgba(230,159,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.lactate,6)","legendgroup":"(basal.lactate,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[2,2],"y":[3.1814350735105,3.26010453126592],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,1)","legendgroup":"(basal.pregnant,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[2,2],"y":[3.390332652155,3.31478960756312],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,2)","legendgroup":"(basal.pregnant,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[2,2],"y":[1.30332802098767,1.40351084187164],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,3)","legendgroup":"(basal.pregnant,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[2,2],"y":[1.35861152420922,1.49697716390167],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,4)","legendgroup":"(basal.pregnant,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[2,2],"y":[2.7126119212013,2.81061305362614],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,5)","legendgroup":"(basal.pregnant,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[2,2],"y":[1.39190821106237,1.72996853097034],"hoverinfo":"y","type":"box","fillcolor":"rgba(86,180,233,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.pregnant,6)","legendgroup":"(basal.pregnant,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[3,3],"y":[3.06271644787554,3.03559456282734],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,1)","legendgroup":"(basal.virgin,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[3,3],"y":[3.11439745253896,3.15755142392862],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,2)","legendgroup":"(basal.virgin,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[3,3],"y":[0.551626182191521,0.750530086982367],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,3)","legendgroup":"(basal.virgin,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[3,3],"y":[1.27715317373435,1.31063344352324],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,4)","legendgroup":"(basal.virgin,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[3,3],"y":[3.1499441563298,3.12682963780376],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,5)","legendgroup":"(basal.virgin,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[3,3],"y":[1.21401331532949,0.816661298761415],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,158,115,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(basal.virgin,6)","legendgroup":"(basal.virgin,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[4,4],"y":[1.5971793558061,1.52687917215815],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,1)","legendgroup":"(luminal.lactate,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[4,4],"y":[4.04774558072317,3.83471665968565],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,2)","legendgroup":"(luminal.lactate,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[4,4],"y":[5.24387148664408,5.23980341109406],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,3)","legendgroup":"(luminal.lactate,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[4,4],"y":[0,0.684864320037247],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,4)","legendgroup":"(luminal.lactate,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[4,4],"y":[3.55976742156017,3.61058611237993],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,5)","legendgroup":"(luminal.lactate,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[4,4],"y":[0.831418551445205,0.938540867662802],"hoverinfo":"y","type":"box","fillcolor":"rgba(240,228,66,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.lactate,6)","legendgroup":"(luminal.lactate,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[5,5],"y":[1.5699654587533,1.61157798837755],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,1)","legendgroup":"(luminal.pregnant,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[5,5],"y":[4.81576284615242,4.97913552489968],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,2)","legendgroup":"(luminal.pregnant,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[5,5],"y":[2.54695989557933,2.76387556513281],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,3)","legendgroup":"(luminal.pregnant,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[5,5],"y":[0.328319501825177,0.351496253591272],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,4)","legendgroup":"(luminal.pregnant,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[5,5],"y":[3.31822770275248,3.30325508377872],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,5)","legendgroup":"(luminal.pregnant,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[5,5],"y":[0.890880865269655,0.777121446515962],"hoverinfo":"y","type":"box","fillcolor":"rgba(0,114,178,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.pregnant,6)","legendgroup":"(luminal.pregnant,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[6,6],"y":[2.19468193165742,2.19469975407936],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,1)","legendgroup":"(luminal.virgin,1)","showlegend":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[6,6],"y":[4.61959639921578,4.69197737763853],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,2)","legendgroup":"(luminal.virgin,2)","showlegend":true,"xaxis":"x2","yaxis":"y","frame":null},{"x":[6,6],"y":[1.88440776492045,1.70078948706894],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,3)","legendgroup":"(luminal.virgin,3)","showlegend":true,"xaxis":"x3","yaxis":"y","frame":null},{"x":[6,6],"y":[0.26938444467675,0.760592085189143],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,4)","legendgroup":"(luminal.virgin,4)","showlegend":true,"xaxis":"x","yaxis":"y2","frame":null},{"x":[6,6],"y":[3.3574931598305,3.3166773115062],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,5)","legendgroup":"(luminal.virgin,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","frame":null},{"x":[6,6],"y":[1.38390765358259,1.36593389291594],"hoverinfo":"y","type":"box","fillcolor":"rgba(213,94,0,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"name":"(luminal.virgin,6)","legendgroup":"(luminal.virgin,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","frame":null},{"x":[0.944558886066079,0.707333443500102],"y":[3.91468147242083,3.82835253986116],"text":["counts_scaled + 1: 8.216345e+03<br />sample: MCL1-DK","counts_scaled + 1: 6.735162e+03<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,1)","legendgroup":"(basal.lactate,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.22875180132687,1.19278551153839],"y":[1.81190806338337,1.71089359173023],"text":["counts_scaled + 1: 6.484983e+01<br />sample: MCL1-DK","counts_scaled + 1: 5.139169e+01<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,2)","legendgroup":"(basal.lactate,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.848202360607684,0.880794011615217],"y":[3.20397604295583,3.14414714113558],"text":["counts_scaled + 1: 1.599486e+03<br />sample: MCL1-DK","counts_scaled + 1: 1.393643e+03<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,3)","legendgroup":"(basal.lactate,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.10588395725936,1.39060011487454],"y":[2.94519882057049,2.99735914130755],"text":["counts_scaled + 1: 8.814555e+02<br />sample: MCL1-DK","counts_scaled + 1: 9.939454e+02<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,4)","legendgroup":"(basal.lactate,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.675300562940538,1.1358627025038],"y":[3.50579150916972,3.4279143804149],"text":["counts_scaled + 1: 3.204693e+03<br />sample: MCL1-DK","counts_scaled + 1: 2.678631e+03<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,5)","legendgroup":"(basal.lactate,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.31806858778,0.646380883082747],"y":[3.39404342058824,3.21904067702014],"text":["counts_scaled + 1: 2.477701e+03<br />sample: MCL1-DK","counts_scaled + 1: 1.655909e+03<br />sample: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.lactate,6)","legendgroup":"(basal.lactate,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.85760035812855,1.75955559723079],"y":[3.18143510280191,3.26010231603762],"text":["counts_scaled + 1: 1.518571e+03<br />sample: MCL1-DI","counts_scaled + 1: 1.820139e+03<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,1)","legendgroup":"(basal.pregnant,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.23936562649906,2.20993234645575],"y":[3.39033448363755,3.31478304417317],"text":["counts_scaled + 1: 2.456590e+03<br />sample: MCL1-DI","counts_scaled + 1: 2.064380e+03<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,2)","legendgroup":"(basal.pregnant,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.88852828294039,1.93035626951605],"y":[1.30333325533803,1.4035125688706],"text":["counts_scaled + 1: 2.010611e+01<br />sample: MCL1-DI","counts_scaled + 1: 2.532275e+01<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,3)","legendgroup":"(basal.pregnant,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.35400429982692,2.25229676291347],"y":[1.35860757867028,1.49698265878697],"text":["counts_scaled + 1: 2.283555e+01<br />sample: MCL1-DI","counts_scaled + 1: 3.140344e+01<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,4)","legendgroup":"(basal.pregnant,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[2.13219178076834,1.93377006370574],"y":[2.71261329587469,2.81061667385325],"text":["counts_scaled + 1: 5.159551e+02<br />sample: MCL1-DI","counts_scaled + 1: 6.465663e+02<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,5)","legendgroup":"(basal.pregnant,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.93367030285299,1.70160223115236],"y":[1.39190138740799,1.72997121646507],"text":["counts_scaled + 1: 2.465518e+01<br />sample: MCL1-DI","counts_scaled + 1: 5.369929e+01<br />sample: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.pregnant,6)","legendgroup":"(basal.pregnant,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.31753623187542,3.21008863300085],"y":[3.06271458109773,3.0355954101102],"text":["counts_scaled + 1: 1.155358e+03<br />sample: MCL1-DG","counts_scaled + 1: 1.085412e+03<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,1)","legendgroup":"(basal.virgin,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.39175421800464,2.81149945501238],"y":[3.11439909970078,3.15755601959577],"text":["counts_scaled + 1: 1.301360e+03<br />sample: MCL1-DG","counts_scaled + 1: 1.437313e+03<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,2)","legendgroup":"(basal.virgin,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.27794266920537,2.88153999913484],"y":[0.551624925428552,0.750536781846052],"text":["counts_scaled + 1: 3.561445e+00<br />sample: MCL1-DG","counts_scaled + 1: 5.630281e+00<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,3)","legendgroup":"(basal.virgin,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.88598809540272,3.24669186007231],"y":[1.27715532033134,1.31063900284507],"text":["counts_scaled + 1: 1.893011e+01<br />sample: MCL1-DG","counts_scaled + 1: 2.044718e+01<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,4)","legendgroup":"(basal.virgin,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[2.73902325965464,3.13617672808468],"y":[3.14994174822357,3.12682270943953],"text":["counts_scaled + 1: 1.412356e+03<br />sample: MCL1-DG","counts_scaled + 1: 1.339151e+03<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,5)","legendgroup":"(basal.virgin,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[2.99258321058005,3.03477658182383],"y":[1.21400898195477,0.816665753022526],"text":["counts_scaled + 1: 1.636867e+01<br />sample: MCL1-DG","counts_scaled + 1: 6.556337e+00<br />sample: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(basal.virgin,6)","legendgroup":"(basal.virgin,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.72913785856217,3.70449563916773],"y":[1.59717805837841,1.52688096879399],"text":["counts_scaled + 1: 3.955299e+01<br />sample: MCL1-LE","counts_scaled + 1: 3.364180e+01<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,1)","legendgroup":"(luminal.lactate,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.68100689444691,3.88047821857035],"y":[4.04774413883452,3.83471646276733],"text":["counts_scaled + 1: 1.116209e+04<br />sample: MCL1-LE","counts_scaled + 1: 6.834656e+03<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,2)","legendgroup":"(luminal.lactate,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.84229677561671,3.64899098463356],"y":[5.24387846238317,5.23980856619432],"text":["counts_scaled + 1: 1.753362e+05<br />sample: MCL1-LE","counts_scaled + 1: 1.737014e+05<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,3)","legendgroup":"(luminal.lactate,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.94430088922381,3.60307067837566],"y":[-3.09400572376989e-06,0.684863195312136],"text":["counts_scaled + 1: 1.000000e+00<br />sample: MCL1-LE","counts_scaled + 1: 4.840211e+00<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,4)","legendgroup":"(luminal.lactate,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.67464352156967,4.26606558933854],"y":[3.55976044995062,3.61058908912509],"text":["counts_scaled + 1: 3.628837e+03<br />sample: MCL1-LE","counts_scaled + 1: 4.079304e+03<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,5)","legendgroup":"(luminal.lactate,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.8772902129218,4.30249013639987],"y":[0.831420617718788,0.938542709196458],"text":["counts_scaled + 1: 6.782949e+00<br />sample: MCL1-LE","counts_scaled + 1: 8.680423e+00<br />sample: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.lactate,6)","legendgroup":"(luminal.lactate,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.92902177702635,4.90322337374091],"y":[1.56996671623181,1.61158224182157],"text":["counts_scaled + 1: 3.715057e+01<br />sample: MCL1-LC","counts_scaled + 1: 4.088632e+01<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,1)","legendgroup":"(luminal.pregnant,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[5.03291758503765,4.81205471660942],"y":[4.81576351034224,4.97913212723169],"text":["counts_scaled + 1: 6.542788e+04<br />sample: MCL1-LC","counts_scaled + 1: 9.530935e+04<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,2)","legendgroup":"(luminal.pregnant,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.77246512379497,4.82846034094691],"y":[2.54695832965884,2.76387864702417],"text":["counts_scaled + 1: 3.523383e+02<br />sample: MCL1-LC","counts_scaled + 1: 5.805980e+02<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,3)","legendgroup":"(luminal.pregnant,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.67799542341381,5.10666911453009],"y":[0.328319450538006,0.351501430485388],"text":["counts_scaled + 1: 2.129705e+00<br />sample: MCL1-LC","counts_scaled + 1: 2.246447e+00<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,4)","legendgroup":"(luminal.pregnant,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.88599276691675,5.21992343533784],"y":[3.31822783906877,3.3032564531319],"text":["counts_scaled + 1: 2.080787e+03<br />sample: MCL1-LC","counts_scaled + 1: 2.010273e+03<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,5)","legendgroup":"(luminal.pregnant,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.9335894972086,5.0459727935493],"y":[0.890874709660555,0.777125484001123],"text":["counts_scaled + 1: 7.778232e+00<br />sample: MCL1-LC","counts_scaled + 1: 5.985790e+00<br />sample: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.pregnant,6)","legendgroup":"(luminal.pregnant,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.79553249608725,5.78549314215779],"y":[2.19467942473375,2.19470434457619],"text":["counts_scaled + 1: 1.565604e+02<br />sample: MCL1-LA","counts_scaled + 1: 1.565668e+02<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,1)","legendgroup":"(luminal.virgin,1)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6.13669853918254,5.61310312785208],"y":[4.61958980337681,4.69197818139105],"text":["counts_scaled + 1: 4.164822e+04<br />sample: MCL1-LA","counts_scaled + 1: 4.920139e+04<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,2)","legendgroup":"(luminal.virgin,2)","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[5.72986460477114,5.69087840374559],"y":[1.88440942391879,1.7007864640071],"text":["counts_scaled + 1: 7.663158e+01<br />sample: MCL1-LA","counts_scaled + 1: 5.020992e+01<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,3)","legendgroup":"(luminal.virgin,3)","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6.28624931424856,5.85216821581125],"y":[0.2693881145683,0.760595868115231],"text":["counts_scaled + 1: 1.859450e+00<br />sample: MCL1-LA","counts_scaled + 1: 5.762250e+00<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,4)","legendgroup":"(luminal.virgin,4)","showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.96955287568271,5.76483562719077],"y":[3.35749062800529,3.31668357522413],"text":["counts_scaled + 1: 2.277682e+03<br />sample: MCL1-LA","counts_scaled + 1: 2.073372e+03<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,5)","legendgroup":"(luminal.virgin,5)","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.900805060938,5.72738127857447],"y":[1.38391077014565,1.36593067463668],"text":["counts_scaled + 1: 2.420514e+01<br />sample: MCL1-LA","counts_scaled + 1: 2.322383e+01<br />sample: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"(luminal.virgin,6)","legendgroup":"(luminal.virgin,6)","showlegend":false,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":40.5768157463462,"r":7.30593607305936,"b":44.1700079422781,"l":58.7795765877958},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xaxis":{"domain":[0,0.32941943900848],"automargin":true,"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-30,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"annotations":[{"text":"Group","x":0.5,"y":-0.0211346180148144,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis"},{"text":"counts_scaled + 1","x":-0.0233054616616261,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"Atp2b4","x":0.16470971950424,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Cldn4","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Csn1s2b","x":0.83529028049576,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Mrgprf","x":0.16470971950424,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Myof","x":0.5,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Tnni2","x":0.83529028049576,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Group","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"yaxis":{"domain":[0.517391429538076,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.262197171825168,5.50607254020261],"tickmode":"array","ticktext":["1e+01","1e+03","1e+05"],"tickvals":[1,3,5],"categoryorder":"array","categoryarray":["1e+01","1e+03","1e+05"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-30,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0.337247227658187,0.662752772341813],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"xaxis3":{"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-30,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0.67058056099152,1],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"yaxis2":{"type":"linear","autorange":false,"range":[-0.262197171825168,5.50607254020261],"tickmode":"array","ticktext":["1e+01","1e+03","1e+05"],"tickvals":[1,3,5],"categoryorder":"array","categoryarray":["1e+01","1e+03","1e+05"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0,0.482608570461924],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":12.7521793275218},"y":0.960675435801739},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"54b837ae16f0":{"x":{},"y":{},"fill":{},"label":{},"type":"box"},"54b83048b23":{"x":{},"y":{},"fill":{},"label":{}}},"cur_data":"54b837ae16f0","visdat":{"54b837ae16f0":["function (y) ","x"],"54b83048b23":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="automatic-bibliography" class="section level2">
<h2 class="hasAnchor">
<a href="#automatic-bibliography" class="anchor"></a>Automatic bibliography</h2>
<p>Tidybulk provides a handy function called <code>get_bibliography</code> that keeps track of the references for the methods used in your tidybulk workflow. The references are in BibTeX format and can be imported into your reference manager.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/get_bibliography-methods.html">get_bibliography</a></span><span class="op">(</span><span class="va">counts_de</span><span class="op">)</span>
<span class="co">#&gt; @article{robinson2010edger,</span>
<span class="co">#&gt;   title={edgeR: a Bioconductor package for differential expression analysis of digital gene expression data},</span>
<span class="co">#&gt;   author={Robinson, Mark D and McCarthy, Davis J and Smyth, Gordon K},</span>
<span class="co">#&gt;   journal={Bioinformatics},</span>
<span class="co">#&gt;   volume={26},</span>
<span class="co">#&gt;   number={1},</span>
<span class="co">#&gt;   pages={139--140},</span>
<span class="co">#&gt;   year={2010},</span>
<span class="co">#&gt;   publisher={Oxford University Press}</span>
<span class="co">#&gt;  }</span>
<span class="co">#&gt; @article{robinson2010scaling,</span>
<span class="co">#&gt;   title={A scaling normalization method for differential expression analysis of RNA-seq data},</span>
<span class="co">#&gt;   author={Robinson, Mark D and Oshlack, Alicia},</span>
<span class="co">#&gt;   journal={Genome biology},</span>
<span class="co">#&gt;   volume={11},</span>
<span class="co">#&gt;   number={3},</span>
<span class="co">#&gt;   pages={1--9},</span>
<span class="co">#&gt;   year={2010},</span>
<span class="co">#&gt;   publisher={BioMed Central}</span>
<span class="co">#&gt;  }</span>
<span class="co">#&gt; @incollection{smyth2005limma,</span>
<span class="co">#&gt;   title={Limma: linear models for microarray data},</span>
<span class="co">#&gt;   author={Smyth, Gordon K},</span>
<span class="co">#&gt;   booktitle={Bioinformatics and computational biology solutions using R and Bioconductor},</span>
<span class="co">#&gt;   pages={397--420},</span>
<span class="co">#&gt;   year={2005},</span>
<span class="co">#&gt;   publisher={Springer}</span>
<span class="co">#&gt;  }</span></code></pre></div>
</div>
<div id="contributing" class="section level1">
<h1 class="hasAnchor">
<a href="#contributing" class="anchor"></a>Contributing</h1>
<p>If you want to suggest improvements for this workshop or ask questions, you can do so as described <a href="https://github.com/mblue9/RNAseq-R-tidyverse/blob/master/CONTRIBUTING.md">here</a>.</p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>Record package and version information with <code>sessionInfo</code></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] tidybulk_1.2.0    tidyHeatmap_1.1.5 GGally_2.0.0      ggrepel_0.8.2    </span>
<span class="co">#&gt;  [5] plotly_4.9.2.1    ggplot2_3.3.2     stringr_1.4.0     readr_1.4.0      </span>
<span class="co">#&gt;  [9] tidyr_1.1.2       dplyr_1.0.2       tibble_3.0.4     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0            rjson_0.2.20               </span>
<span class="co">#&gt;   [3] ellipsis_0.3.1              ggridges_0.5.2             </span>
<span class="co">#&gt;   [5] rprojroot_2.0.2             circlize_0.4.11            </span>
<span class="co">#&gt;   [7] XVector_0.30.0              GenomicRanges_1.42.0       </span>
<span class="co">#&gt;   [9] GlobalOptions_0.1.2         fs_1.5.0                   </span>
<span class="co">#&gt;  [11] clue_0.3-58                 rstudioapi_0.13            </span>
<span class="co">#&gt;  [13] farver_2.0.3                bit64_4.0.5                </span>
<span class="co">#&gt;  [15] AnnotationDbi_1.52.0        fansi_0.4.1                </span>
<span class="co">#&gt;  [17] splines_4.0.3               geneplotter_1.68.0         </span>
<span class="co">#&gt;  [19] knitr_1.30                  RNAseqRtidyverse_0.0.1     </span>
<span class="co">#&gt;  [21] tidygate_0.3.2              jsonlite_1.7.1             </span>
<span class="co">#&gt;  [23] Cairo_1.5-12.2              annotate_1.68.0            </span>
<span class="co">#&gt;  [25] cluster_2.1.0               png_0.1-7                  </span>
<span class="co">#&gt;  [27] pheatmap_1.0.12             compiler_4.0.3             </span>
<span class="co">#&gt;  [29] httr_1.4.2                  assertthat_0.2.1           </span>
<span class="co">#&gt;  [31] Matrix_1.2-18               lazyeval_0.2.2             </span>
<span class="co">#&gt;  [33] limma_3.46.0                cli_2.2.0                  </span>
<span class="co">#&gt;  [35] org.Mm.eg.db_3.12.0         htmltools_0.5.0            </span>
<span class="co">#&gt;  [37] tools_4.0.3                 gtable_0.3.0               </span>
<span class="co">#&gt;  [39] glue_1.4.2                  GenomeInfoDbData_1.2.4     </span>
<span class="co">#&gt;  [41] Rcpp_1.0.5                  Biobase_2.50.0             </span>
<span class="co">#&gt;  [43] pkgdown_1.6.1               vctrs_0.3.5                </span>
<span class="co">#&gt;  [45] preprocessCore_1.52.0       crosstalk_1.1.0.1          </span>
<span class="co">#&gt;  [47] xfun_0.19                   ps_1.5.0                   </span>
<span class="co">#&gt;  [49] lifecycle_0.2.0             XML_3.99-0.5               </span>
<span class="co">#&gt;  [51] edgeR_3.32.0                zlibbioc_1.36.0            </span>
<span class="co">#&gt;  [53] scales_1.1.1                ragg_0.4.0                 </span>
<span class="co">#&gt;  [55] hms_0.5.3                   MatrixGenerics_1.2.0       </span>
<span class="co">#&gt;  [57] parallel_4.0.3              SummarizedExperiment_1.20.0</span>
<span class="co">#&gt;  [59] RColorBrewer_1.1-2          SingleCellExperiment_1.12.0</span>
<span class="co">#&gt;  [61] ComplexHeatmap_2.6.2        yaml_2.2.1                 </span>
<span class="co">#&gt;  [63] curl_4.3                    memoise_1.1.0              </span>
<span class="co">#&gt;  [65] gridExtra_2.3               reshape_0.8.8              </span>
<span class="co">#&gt;  [67] stringi_1.5.3               RSQLite_2.2.1              </span>
<span class="co">#&gt;  [69] genefilter_1.72.0           S4Vectors_0.28.0           </span>
<span class="co">#&gt;  [71] desc_1.2.0                  BiocGenerics_0.36.0        </span>
<span class="co">#&gt;  [73] BiocParallel_1.24.1         shape_1.4.5                </span>
<span class="co">#&gt;  [75] GenomeInfoDb_1.26.1         rlang_0.4.9                </span>
<span class="co">#&gt;  [77] pkgconfig_2.0.3             systemfonts_0.3.2          </span>
<span class="co">#&gt;  [79] matrixStats_0.57.0          bitops_1.0-6               </span>
<span class="co">#&gt;  [81] evaluate_0.14               lattice_0.20-41            </span>
<span class="co">#&gt;  [83] purrr_0.3.4                 htmlwidgets_1.5.2          </span>
<span class="co">#&gt;  [85] labeling_0.4.2              cowplot_1.1.0              </span>
<span class="co">#&gt;  [87] bit_4.0.4                   tidyselect_1.1.0           </span>
<span class="co">#&gt;  [89] plyr_1.8.6                  magrittr_2.0.1             </span>
<span class="co">#&gt;  [91] DESeq2_1.30.0               R6_2.5.0                   </span>
<span class="co">#&gt;  [93] IRanges_2.24.0              generics_0.1.0             </span>
<span class="co">#&gt;  [95] DelayedArray_0.16.0         DBI_1.1.0                  </span>
<span class="co">#&gt;  [97] pillar_1.4.7                withr_2.3.0                </span>
<span class="co">#&gt;  [99] survival_3.2-7              RCurl_1.98-1.2             </span>
<span class="co">#&gt; [101] crayon_1.3.4                utf8_1.1.4                 </span>
<span class="co">#&gt; [103] dittoSeq_1.2.2              rmarkdown_2.5              </span>
<span class="co">#&gt; [105] viridis_0.5.1               GetoptLong_1.0.4           </span>
<span class="co">#&gt; [107] locfit_1.5-9.4              grid_4.0.3                 </span>
<span class="co">#&gt; [109] data.table_1.13.4           blob_1.2.1                 </span>
<span class="co">#&gt; [111] digest_0.6.27               xtable_1.8-4               </span>
<span class="co">#&gt; [113] textshaping_0.2.1           stats4_4.0.3               </span>
<span class="co">#&gt; [115] munsell_0.5.0               viridisLite_0.3.0</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-chen2016reads">
<p>Chen, Yunshun, Aaron TL Lun, and Gordon K Smyth. 2016. “From Reads to Genes to Pathways: Differential Expression Analysis of Rna-Seq Experiments Using Rsubread and the edgeR Quasi-Likelihood Pipeline.” <em>F1000Research</em> 5.</p>
</div>
<div id="ref-law2016rna">
<p>Law, Charity W, Monther Alhamdoosh, Shian Su, Xueyi Dong, Luyi Tian, Gordon K Smyth, and Matthew E Ritchie. 2016. “RNA-Seq Analysis Is Easy as 1-2-3 with Limma, Glimma and edgeR.” <em>F1000Research</em> 5.</p>
</div>
<div id="ref-law2014voom">
<p>Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. “Voom: Precision Weights Unlock Linear Model Analysis Tools for Rna-Seq Read Counts.” <em>Genome Biology</em> 15 (2): R29.</p>
</div>
<div id="ref-love2014moderated">
<p>Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12): 550.</p>
</div>
<div id="ref-mangiola2020tidyheatmap">
<p>Mangiola, Stefano, and Anthony T Papenfuss. 2020. “TidyHeatmap: An R Package for Modular Heatmap Production Based on Tidy Principles.” <em>Journal of Open Source Software</em> 5 (52): 2472.</p>
</div>
<div id="ref-mccarthy2012differential">
<p>McCarthy, Davis J, Yunshun Chen, and Gordon K Smyth. 2012. “Differential Expression Analysis of Multifactor Rna-Seq Experiments with Respect to Biological Variation.” <em>Nucleic Acids Research</em> 40 (10): 4288–97.</p>
</div>
<div id="ref-robinson2010scaling">
<p>Robinson, Mark D, and Alicia Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of Rna-Seq Data.” <em>Genome Biology</em> 11 (3): 1–9.</p>
</div>
<div id="ref-wickham2019welcome">
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the Tidyverse.” <em>Journal of Open Source Software</em> 4 (43): 1686.</p>
</div>
<div id="ref-wickham2014tidy">
<p>Wickham, Hadley, and others. 2014. “Tidy Data.” <em>Journal of Statistical Software</em> 59 (10): 1–23.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>&lt;maria.doyle at petermac.org&gt;<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>&lt;mangiola.s at wehi.edu.au&gt;<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Doyle, Stefano Mangiola.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
