<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RNAseq-R-tidyverse • RNAseqRtidyverse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RNAseq-R-tidyverse">
<meta property="og:description" content="RNAseqRtidyverse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RNAseqRtidyverse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/tidytranscriptomics.html">Material</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mblue9/RNAseq-R-tidyverse">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://hub.docker.com/repository/docker/mblue9/RNAseq-R-tidyverse">
    <span class="fab fa-docker"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tidytranscriptomics_files/header-attrs-2.9/header-attrs.js"></script><script src="tidytranscriptomics_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="tidytranscriptomics_files/plotly-binding-4.9.4.1/plotly.js"></script><script src="tidytranscriptomics_files/typedarray-0.1/typedarray.min.js"></script><script src="tidytranscriptomics_files/jquery-3.5.1/jquery.min.js"></script><link href="tidytranscriptomics_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="tidytranscriptomics_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><link href="tidytranscriptomics_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="tidytranscriptomics_files/plotly-main-1.57.1/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RNAseq-R-tidyverse</h1>
                        <h4 class="author">Maria Doyle, Peter MacCallum Cancer Centre<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">Stefano Mangiola, Walter and Eliza Hall Institute<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            
            <h4 class="date">14 July 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mblue9/RNAseq-R-tidyverse/blob/master/vignettes/tidytranscriptomics.Rmd"><code>vignettes/tidytranscriptomics.Rmd</code></a></small>
      <div class="hidden name"><code>tidytranscriptomics.Rmd</code></div>

    </div>

    
    
<p>This material presents how to perform analysis of RNA sequencing data following the tidy data paradigm <span class="citation">[@wickham2014tidy]</span>. The tidy data paradigm provides a standard way to organise data values within a dataset, where each variable is a column, each observation is a row, and data is manipulated using an easy-to-understand vocabulary. Most importantly, the data structure remains consistent across manipulation and analysis functions.</p>
<p>This can be achieved for RNA sequencing data with the <a href="https://stemangiola.github.io/tidybulk/">tidybulk</a><span class="citation">[@Mangiola2021]</span>, <a href="https://stemangiola.github.io/tidyHeatmap">tidyHeatmap</a> <span class="citation">[@mangiola2020tidyheatmap]</span> and <a href="https://www.tidyverse.org/">tidyverse</a> <span class="citation">[@wickham2019welcome]</span> packages. The tidybulk package provides a tidy data structure and a modular framework for bulk transcriptional analyses. tidyHeatmap provides a tidy implementation of ComplexHeatmap. These packages are part of the tidytranscriptomics suite that introduces a tidy approach to RNA sequencing data representation and analysis</p>
<div id="acknowledgements" class="section level3">
<h3 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h3>
<p>This material was adapted from an R for RNA sequencing workshop first run <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">here</a>.</p>
<p><img src="../inst/vignettes/tidybulk_logo.png" width="100px"></p>
</div>
<div id="introduction-and-data-import" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-and-data-import" class="anchor"></a>Introduction and data import</h2>
<p>Measuring gene expression on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene expression in often complex systems. As sequencing costs have decreased, using RNA sequencing to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.</p>
<p>There are many steps involved in analysing an RNA sequencing dataset. Sequenced reads are aligned to a reference genome, then the number of reads mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on in R. While mapping and counting are important and necessary tasks, today we will be starting from the count data and showing how differential expression analysis can be performed in a friendly way using the Bioconductor package, tidybulk.</p>
<p>First, let’s load all the packages we will need to analyse the data.</p>
<p><em>Note: you should load the <em>tidybulk</em> library after the tidyverse core packages for best integration.</em></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load libraries</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/">tibble</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org">readr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com">plotly</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">EGSEA</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.r-project.org">tidyHeatmap</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">tidybulk</span><span class="op">)</span></code></pre></div>
<p>Plot settings. Set the colours and theme we will use for our plots.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use colourblind-friendly colours</span>
<span class="va">friendly_cols</span> <span class="op">&lt;-</span> <span class="fu">dittoSeq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dittoSeq/man/dittoColors.html">dittoColors</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Set theme</span>
<span class="va">custom_theme</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">friendly_cols</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">friendly_cols</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
        panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        axis.line <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
        panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,
        text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">10</span>, r <span class="op">=</span> <span class="fl">10</span>, b <span class="op">=</span> <span class="fl">10</span>, l <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">10</span>, r <span class="op">=</span> <span class="fl">10</span>, b <span class="op">=</span> <span class="fl">10</span>, l <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">30</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
      <span class="op">)</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="mouse-mammary-gland-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#mouse-mammary-gland-dataset" class="anchor"></a>Mouse mammary gland dataset</h2>
<p>Here we will perform RNA-Seq analysis using data from a breast cancer research study, from the paper by <a href="https://www.ncbi.nlm.nih.gov/pubmed/25730472">Fu et al. 2015</a>, GEO code GSE60450. This study examined gene expression in basal and luminal cells from mice at different stages of mammary gland development (virgin, pregnant and lactating). There are 2 samples per group and 6 groups, 12 samples in total.</p>
</div>
<div id="reading-in-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-in-the-data" class="anchor"></a>Reading in the data</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># import RNA-seq counts</span>
<span class="va">seqdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"https://ndownloader.figshare.com/files/5057929?private_link=1d788fd384d33e913a2a"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   EntrezGeneID = col_double(),</span>
<span class="co">#&gt;   Length = col_double(),</span>
<span class="co">#&gt;   `MCL1-DG_BC2CTUACXX_ACTTGA_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DH_BC2CTUACXX_CAGATC_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DI_BC2CTUACXX_ACAGTG_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-DL_BC2CTUACXX_ATCACG_L002_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LA_BC2CTUACXX_GATCAG_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LB_BC2CTUACXX_TGACCA_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LE_BC2CTUACXX_TAGCTT_L001_R1` = col_double(),</span>
<span class="co">#&gt;   `MCL1-LF_BC2CTUACXX_CTTGTA_L001_R1` = col_double()</span>
<span class="co">#&gt; )</span>

<span class="co"># import sample information</span>
<span class="va">sampleinfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"https://ndownloader.figshare.com/files/5999832?private_link=1d788fd384d33e913a2a"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   FileName = col_character(),</span>
<span class="co">#&gt;   SampleName = col_character(),</span>
<span class="co">#&gt;   CellType = col_character(),</span>
<span class="co">#&gt;   Status = col_character()</span>
<span class="co">#&gt; )</span></code></pre></div>
<p>Take a look at the data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqdata</span>
<span class="co">#&gt; # A tibble: 27,179 x 14</span>
<span class="co">#&gt;    EntrezGeneID Length `MCL1-DG_BC2CTUACX… `MCL1-DH_BC2CTUAC… `MCL1-DI_BC2CTUAC…</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt;               &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;</span>
<span class="co">#&gt;  1       497097   3634                 438                300                 65</span>
<span class="co">#&gt;  2    100503874   3259                   1                  0                  1</span>
<span class="co">#&gt;  3    100038431   1634                   0                  0                  0</span>
<span class="co">#&gt;  4        19888   9747                   1                  1                  0</span>
<span class="co">#&gt;  5        20671   3130                 106                182                 82</span>
<span class="co">#&gt;  6        27395   4203                 309                234                337</span>
<span class="co">#&gt;  7        18777   2433                 652                515                948</span>
<span class="co">#&gt;  8    100503730    799                   0                  1                  0</span>
<span class="co">#&gt;  9        21399   2847                1604               1495               1721</span>
<span class="co">#&gt; 10        58175   2241                   4                  2                 14</span>
<span class="co">#&gt; # … with 27,169 more rows, and 9 more variables:</span>
<span class="co">#&gt; #   MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-DL_BC2CTUACXX_ATCACG_L002_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LA_BC2CTUACXX_GATCAG_L001_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LB_BC2CTUACXX_TGACCA_L001_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LE_BC2CTUACXX_TAGCTT_L001_R1 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   MCL1-LF_BC2CTUACXX_CTTGTA_L001_R1 &lt;dbl&gt;</span></code></pre></div>
<p>In the <code>seqdata</code> object, the first column contains the Entrez gene identifier, the second column contains the gene length and the rest of the columns contain the gene transcription abundance for each sample. The abundance is the number of reads aligning to the gene in each experimental sample.</p>
</div>
<div id="format-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#format-the-data" class="anchor"></a>Format the data</h2>
<p>We first convert the counts into long format (tidy format) so we can join to the sample information and also use with ggplot and other tidyverse functions.</p>
<p>In this workshop we make use of the tidyverse pipe <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code>. This ‘pipes’ the output from the command on the left into the command on the right/below. Using the pipe is not essential but it can make the steps clearer and easier to see. For more details on the pipe see <a href="https://r4ds.had.co.nz/pipes.html">here</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_long</span> <span class="op">&lt;-</span> <span class="va">seqdata</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"MCL"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"sample"</span>, values_to <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_long</span>
<span class="co">#&gt; # A tibble: 326,148 x 4</span>
<span class="co">#&gt;    EntrezGeneID Length sample                            counts</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                              &lt;dbl&gt;</span>
<span class="co">#&gt;  1       497097   3634 MCL1-DG_BC2CTUACXX_ACTTGA_L002_R1    438</span>
<span class="co">#&gt;  2       497097   3634 MCL1-DH_BC2CTUACXX_CAGATC_L002_R1    300</span>
<span class="co">#&gt;  3       497097   3634 MCL1-DI_BC2CTUACXX_ACAGTG_L002_R1     65</span>
<span class="co">#&gt;  4       497097   3634 MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1    237</span>
<span class="co">#&gt;  5       497097   3634 MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1    354</span>
<span class="co">#&gt;  6       497097   3634 MCL1-DL_BC2CTUACXX_ATCACG_L002_R1    287</span>
<span class="co">#&gt;  7       497097   3634 MCL1-LA_BC2CTUACXX_GATCAG_L001_R1      0</span>
<span class="co">#&gt;  8       497097   3634 MCL1-LB_BC2CTUACXX_TGACCA_L001_R1      0</span>
<span class="co">#&gt;  9       497097   3634 MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1      0</span>
<span class="co">#&gt; 10       497097   3634 MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1      0</span>
<span class="co">#&gt; # … with 326,138 more rows</span></code></pre></div>
<p>We can shorten the sample names to just the relevant information, the first 7 characters.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_formatted</span> <span class="op">&lt;-</span> <span class="va">counts_long</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sample_name<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html">str_extract</a></span><span class="op">(</span><span class="va">sample</span>, <span class="st">"MCL1-.."</span><span class="op">)</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_formatted</span>
<span class="co">#&gt; # A tibble: 326,148 x 5</span>
<span class="co">#&gt;    EntrezGeneID Length sample                            counts sample_name</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                              &lt;dbl&gt; &lt;chr&gt;      </span>
<span class="co">#&gt;  1       497097   3634 MCL1-DG_BC2CTUACXX_ACTTGA_L002_R1    438 MCL1-DG    </span>
<span class="co">#&gt;  2       497097   3634 MCL1-DH_BC2CTUACXX_CAGATC_L002_R1    300 MCL1-DH    </span>
<span class="co">#&gt;  3       497097   3634 MCL1-DI_BC2CTUACXX_ACAGTG_L002_R1     65 MCL1-DI    </span>
<span class="co">#&gt;  4       497097   3634 MCL1-DJ_BC2CTUACXX_CGATGT_L002_R1    237 MCL1-DJ    </span>
<span class="co">#&gt;  5       497097   3634 MCL1-DK_BC2CTUACXX_TTAGGC_L002_R1    354 MCL1-DK    </span>
<span class="co">#&gt;  6       497097   3634 MCL1-DL_BC2CTUACXX_ATCACG_L002_R1    287 MCL1-DL    </span>
<span class="co">#&gt;  7       497097   3634 MCL1-LA_BC2CTUACXX_GATCAG_L001_R1      0 MCL1-LA    </span>
<span class="co">#&gt;  8       497097   3634 MCL1-LB_BC2CTUACXX_TGACCA_L001_R1      0 MCL1-LB    </span>
<span class="co">#&gt;  9       497097   3634 MCL1-LC_BC2CTUACXX_GCCAAT_L001_R1      0 MCL1-LC    </span>
<span class="co">#&gt; 10       497097   3634 MCL1-LD_BC2CTUACXX_GGCTAC_L001_R1      0 MCL1-LD    </span>
<span class="co">#&gt; # … with 326,138 more rows</span></code></pre></div>
<p>Now let’s take a look at the sample information table.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleinfo</span>
<span class="co">#&gt; # A tibble: 12 x 4</span>
<span class="co">#&gt;    FileName                          SampleName CellType Status  </span>
<span class="co">#&gt;    &lt;chr&gt;                             &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;   </span>
<span class="co">#&gt;  1 MCL1.DG_BC2CTUACXX_ACTTGA_L002_R1 MCL1.DG    basal    virgin  </span>
<span class="co">#&gt;  2 MCL1.DH_BC2CTUACXX_CAGATC_L002_R1 MCL1.DH    basal    virgin  </span>
<span class="co">#&gt;  3 MCL1.DI_BC2CTUACXX_ACAGTG_L002_R1 MCL1.DI    basal    pregnant</span>
<span class="co">#&gt;  4 MCL1.DJ_BC2CTUACXX_CGATGT_L002_R1 MCL1.DJ    basal    pregnant</span>
<span class="co">#&gt;  5 MCL1.DK_BC2CTUACXX_TTAGGC_L002_R1 MCL1.DK    basal    lactate </span>
<span class="co">#&gt;  6 MCL1.DL_BC2CTUACXX_ATCACG_L002_R1 MCL1.DL    basal    lactate </span>
<span class="co">#&gt;  7 MCL1.LA_BC2CTUACXX_GATCAG_L001_R1 MCL1.LA    luminal  virgin  </span>
<span class="co">#&gt;  8 MCL1.LB_BC2CTUACXX_TGACCA_L001_R1 MCL1.LB    luminal  virgin  </span>
<span class="co">#&gt;  9 MCL1.LC_BC2CTUACXX_GCCAAT_L001_R1 MCL1.LC    luminal  pregnant</span>
<span class="co">#&gt; 10 MCL1.LD_BC2CTUACXX_GGCTAC_L001_R1 MCL1.LD    luminal  pregnant</span>
<span class="co">#&gt; 11 MCL1.LE_BC2CTUACXX_TAGCTT_L001_R1 MCL1.LE    luminal  lactate </span>
<span class="co">#&gt; 12 MCL1.LF_BC2CTUACXX_CTTGTA_L001_R1 MCL1.LF    luminal  lactate</span></code></pre></div>
<p>The CellType column tells us whether the samples are from basal or luminal cells and the Status column tells us what stage of mouse mammary gland development they are from.</p>
<p>We’ll create a column containing the groups we’ll be examining e.g. basal.pregnant and basal.lactate by joining the CellType and Status columns using <code>unite</code>. We’ll use <code>mutate</code> to edit the sample id so we can join to the counts.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleinfo_formatted</span> <span class="op">&lt;-</span> <span class="va">sampleinfo</span> <span class="op">%&gt;%</span>
    
    <span class="co"># make column called Group by combining the CellType and Status columns</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html">unite</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="va">CellType</span><span class="op">:</span><span class="va">Status</span>, sep<span class="op">=</span><span class="st">"."</span>, remove<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
    <span class="co"># replace the . in the SampleName with - so can join to counts</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sample_name<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">SampleName</span>, <span class="st">"\\."</span>, <span class="st">"-"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">sampleinfo_formatted</span> 
<span class="co">#&gt; # A tibble: 12 x 6</span>
<span class="co">#&gt;    FileName                 SampleName Group        CellType Status  sample_name</span>
<span class="co">#&gt;    &lt;chr&gt;                    &lt;chr&gt;      &lt;chr&gt;        &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;      </span>
<span class="co">#&gt;  1 MCL1.DG_BC2CTUACXX_ACTT… MCL1.DG    basal.virgin basal    virgin  MCL1-DG    </span>
<span class="co">#&gt;  2 MCL1.DH_BC2CTUACXX_CAGA… MCL1.DH    basal.virgin basal    virgin  MCL1-DH    </span>
<span class="co">#&gt;  3 MCL1.DI_BC2CTUACXX_ACAG… MCL1.DI    basal.pregn… basal    pregna… MCL1-DI    </span>
<span class="co">#&gt;  4 MCL1.DJ_BC2CTUACXX_CGAT… MCL1.DJ    basal.pregn… basal    pregna… MCL1-DJ    </span>
<span class="co">#&gt;  5 MCL1.DK_BC2CTUACXX_TTAG… MCL1.DK    basal.lacta… basal    lactate MCL1-DK    </span>
<span class="co">#&gt;  6 MCL1.DL_BC2CTUACXX_ATCA… MCL1.DL    basal.lacta… basal    lactate MCL1-DL    </span>
<span class="co">#&gt;  7 MCL1.LA_BC2CTUACXX_GATC… MCL1.LA    luminal.vir… luminal  virgin  MCL1-LA    </span>
<span class="co">#&gt;  8 MCL1.LB_BC2CTUACXX_TGAC… MCL1.LB    luminal.vir… luminal  virgin  MCL1-LB    </span>
<span class="co">#&gt;  9 MCL1.LC_BC2CTUACXX_GCCA… MCL1.LC    luminal.pre… luminal  pregna… MCL1-LC    </span>
<span class="co">#&gt; 10 MCL1.LD_BC2CTUACXX_GGCT… MCL1.LD    luminal.pre… luminal  pregna… MCL1-LD    </span>
<span class="co">#&gt; 11 MCL1.LE_BC2CTUACXX_TAGC… MCL1.LE    luminal.lac… luminal  lactate MCL1-LE    </span>
<span class="co">#&gt; 12 MCL1.LF_BC2CTUACXX_CTTG… MCL1.LF    luminal.lac… luminal  lactate MCL1-LF</span></code></pre></div>
</div>
<div id="adding-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-annotation" class="anchor"></a>Adding annotation</h2>
<p>To the counts table we can add gene symbols.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_annotated</span> <span class="op">&lt;-</span> <span class="va">counts_formatted</span> <span class="op">%&gt;%</span>
    
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Mm.eg.db/man/org.Mm.egBASE.html">org.Mm.eg.db</a></span>, 
                                          keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">EntrezGeneID</span><span class="op">)</span>, 
                                          keytype <span class="op">=</span> <span class="st">"ENTREZID"</span>, 
                                          column<span class="op">=</span><span class="st">"SYMBOL"</span>, 
                                          multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co"># take a look</span>
<span class="va">counts_annotated</span>
<span class="co">#&gt; # A tibble: 326,148 x 6</span>
<span class="co">#&gt;    EntrezGeneID Length sample                          counts sample_name symbol</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                            &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt; </span>
<span class="co">#&gt;  1       497097   3634 MCL1-DG_BC2CTUACXX_ACTTGA_L002…    438 MCL1-DG     Xkr4  </span>
<span class="co">#&gt;  2       497097   3634 MCL1-DH_BC2CTUACXX_CAGATC_L002…    300 MCL1-DH     Xkr4  </span>
<span class="co">#&gt;  3       497097   3634 MCL1-DI_BC2CTUACXX_ACAGTG_L002…     65 MCL1-DI     Xkr4  </span>
<span class="co">#&gt;  4       497097   3634 MCL1-DJ_BC2CTUACXX_CGATGT_L002…    237 MCL1-DJ     Xkr4  </span>
<span class="co">#&gt;  5       497097   3634 MCL1-DK_BC2CTUACXX_TTAGGC_L002…    354 MCL1-DK     Xkr4  </span>
<span class="co">#&gt;  6       497097   3634 MCL1-DL_BC2CTUACXX_ATCACG_L002…    287 MCL1-DL     Xkr4  </span>
<span class="co">#&gt;  7       497097   3634 MCL1-LA_BC2CTUACXX_GATCAG_L001…      0 MCL1-LA     Xkr4  </span>
<span class="co">#&gt;  8       497097   3634 MCL1-LB_BC2CTUACXX_TGACCA_L001…      0 MCL1-LB     Xkr4  </span>
<span class="co">#&gt;  9       497097   3634 MCL1-LC_BC2CTUACXX_GCCAAT_L001…      0 MCL1-LC     Xkr4  </span>
<span class="co">#&gt; 10       497097   3634 MCL1-LD_BC2CTUACXX_GGCTAC_L001…      0 MCL1-LD     Xkr4  </span>
<span class="co">#&gt; # … with 326,138 more rows</span></code></pre></div>
<p>We join the counts and sample information into one table.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">counts_annotated</span>, <span class="va">sampleinfo_formatted</span>, by <span class="op">=</span> <span class="st">"sample_name"</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts</span>
<span class="co">#&gt; # A tibble: 326,148 x 11</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;           &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1       497097   3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2       497097   3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3       497097   3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4       497097   3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5       497097   3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6       497097   3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7       497097   3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8       497097   3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9       497097   3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10       497097   3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 326,138 more rows, and 3 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;</span></code></pre></div>
</div>
<div id="convert-counts-to-tidybulk-tibble" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-counts-to-tidybulk-tibble" class="anchor"></a>Convert counts to tidybulk tibble</h2>
<p>To perform the RNA sequencing analysis steps, we convert into a tidybulk tibble. A tibble is the tidyverse table format.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_tt</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span> 
    <span class="co"># convert ids from numeric to character (text) datatype</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>EntrezGeneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">EntrezGeneID</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/tidybulk-methods.html">tidybulk</a></span><span class="op">(</span>.sample<span class="op">=</span><span class="va">sample_name</span>, .transcript<span class="op">=</span><span class="va">EntrezGeneID</span>, .abundance<span class="op">=</span><span class="va">counts</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_tt</span>
<span class="co">#&gt; # A tibble: 326,148 x 11</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1 497097         3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2 497097         3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3 497097         3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4 497097         3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5 497097         3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6 497097         3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7 497097         3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8 497097         3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9 497097         3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10 497097         3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 326,138 more rows, and 3 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;</span></code></pre></div>
<p>Using this tidybulk tibble we can perform differential expression analysis with the tidybulk package.</p>
</div>
<div id="filtering-lowly-expressed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-lowly-expressed-genes" class="anchor"></a>Filtering lowly expressed genes</h2>
<p>Genes with very low counts across all libraries provide little evidence for differential expression and they can interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes should be filtered out prior to further analysis.</p>
<p>We can perform the filtering using tidybulk <code>keep_abundant</code> or <code>identify_abundant</code>. These functions can use the <em>edgeR</em> <code>filterByExpr</code> function described in <span class="citation">[@law2016rna]</span> to automatically identify the genes with adequate abundance for differential expression testing. By default, this will keep genes with ~10 counts in a minimum number of samples, the number of the samples in the smallest group. In this dataset the smallest group size is two (we have two replicates for each group). Alternatively, we could use <code>identify_abundant</code> to identify which genes are abundant or not (TRUE/FALSE), rather than just keeping the abundant ones.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filtering counts</span>
<span class="va">counts_filtered</span> <span class="op">&lt;-</span> <span class="va">counts_tt</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/keep_abundant-methods.html">keep_abundant</a></span><span class="op">(</span>factor_of_interest <span class="op">=</span> <span class="va">Group</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_filtered</span>
<span class="co">#&gt; # A tibble: 191,628 x 12</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1 497097         3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2 497097         3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3 497097         3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4 497097         3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5 497097         3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6 497097         3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7 497097         3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8 497097         3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9 497097         3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10 497097         3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 191,618 more rows, and 4 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;, .abundant &lt;lgl&gt;</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take a look</span>
<span class="va">counts_filtered</span>
<span class="co">#&gt; # A tibble: 191,628 x 12</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1 497097         3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2 497097         3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3 497097         3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4 497097         3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5 497097         3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6 497097         3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7 497097         3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8 497097         3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9 497097         3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10 497097         3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 191,618 more rows, and 4 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;, .abundant &lt;lgl&gt;</span></code></pre></div>
<p>After running <code>keep_abundant</code> we have a column called <code>.abundant</code> containing TRUE (<code>identify_abundant</code> would have TRUE/FALSE).</p>
</div>
<div id="quality-control" class="section level1">
<h1 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h1>
<p>Now that we have got rid of the lowly expressed genes, we can look at a few different plots to check that the data is good quality, and that the samples are as we would expect.</p>
<div id="library-size-and-distribution-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#library-size-and-distribution-plots" class="anchor"></a>Library size and distribution plots</h2>
<p>First, we can check how many reads we have for each sample.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_tt</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>total_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 x 2</span>
<span class="co">#&gt;    sample_name total_reads</span>
<span class="co">#&gt;    &lt;chr&gt;             &lt;dbl&gt;</span>
<span class="co">#&gt;  1 MCL1-DG        23227641</span>
<span class="co">#&gt;  2 MCL1-DH        21777891</span>
<span class="co">#&gt;  3 MCL1-DI        24100765</span>
<span class="co">#&gt;  4 MCL1-DJ        22665371</span>
<span class="co">#&gt;  5 MCL1-DK        21529331</span>
<span class="co">#&gt;  6 MCL1-DL        20015386</span>
<span class="co">#&gt;  7 MCL1-LA        20392113</span>
<span class="co">#&gt;  8 MCL1-LB        21708152</span>
<span class="co">#&gt;  9 MCL1-LC        22241607</span>
<span class="co">#&gt; 10 MCL1-LD        21988240</span>
<span class="co">#&gt; 11 MCL1-LE        24723827</span>
<span class="co">#&gt; 12 MCL1-LF        24657293</span></code></pre></div>
<p>We can also plot the library sizes as a barplot to see whether there are any major discrepancies between the samples more easily.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">counts_tt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample_name</span>, weight <span class="op">=</span> <span class="va">counts</span>, fill <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
</div>
<div id="scaling-counts-to-normalise" class="section level2">
<h2 class="hasAnchor">
<a href="#scaling-counts-to-normalise" class="anchor"></a>Scaling counts to normalise</h2>
<p>Scaling of counts, normalisation, is performed to eliminate uninteresting differences between samples due to sequencing depth or composition. A more detailed explanation can be found <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>. In the tidybulk package the function <code>scale_abundance</code> generates scaled counts, with scaling factors calculated on abundant (filtered) transcripts and applied to all transcripts. We can choose from different normalisation methods. Here we will use the default, edgeR’s trimmed mean of M values (TMM), <span class="citation">[@robinson2010scaling]</span>. TMM normalisation (and most scaling normalisation methods) scale relative to one sample.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Scaling counts</span>
<span class="va">counts_scaled</span> <span class="op">&lt;-</span> <span class="va">counts_filtered</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/scale_abundance-methods.html">scale_abundance</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># take a look</span>
<span class="va">counts_scaled</span>
<span class="co">#&gt; # A tibble: 191,628 x 15</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1 497097         3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2 497097         3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3 497097         3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4 497097         3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5 497097         3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6 497097         3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7 497097         3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8 497097         3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9 497097         3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10 497097         3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 191,618 more rows, and 7 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;, .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;, multiplier &lt;dbl&gt;,</span>
<span class="co">#&gt; #   counts_scaled &lt;dbl&gt;</span></code></pre></div>
<p>After we run <code>scale_abundance</code> we should see some columns have been added at the end. The <code>counts_scaled</code> column contains the scaled counts.</p>
<p>We can visualise the difference of abundance distributions before and after scaling. As tidybulk output is compatible with tidyverse, we can simply pipe it into standard tidyverse functions such as <code>filter</code>, <code>pivot_longer</code> and <code>ggplot</code>. We can also take advantage of ggplot’s <code>facet_wrap</code> to easily create multiple plots.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_scaled</span> <span class="op">%&gt;%</span>

  <span class="co"># Reshaping        </span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"counts"</span>, <span class="st">"counts_scaled"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"source"</span>, values_to <span class="op">=</span> <span class="st">"abundance"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
  <span class="co"># Plotting</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample_name</span>, y <span class="op">=</span> <span class="va">abundance</span> <span class="op">+</span> <span class="fl">1</span>, fill <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">abundance</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">source</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-19-1.png" width="70%"></p>
<p>In this dataset the distributions of the counts are not very different to each other before scaling but scaling does make the distributions more similar. If we saw a sample with a very different distribution we may need to investigate it.</p>
</div>
<div id="exploratory-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-analyses" class="anchor"></a>Exploratory analyses</h2>
<div id="dimensionality-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#dimensionality-reduction" class="anchor"></a>Dimensionality reduction</h3>
<p>By far, one of the most important plots we make when we analyse RNA sequencing data are principal-component analysis (PCA) or multi-dimensional scaling (MDS) plots. We reduce the dimensions of the data to identify the greatest sources of variation in the data. A principal components analysis is an example of an unsupervised analysis, where we don’t need to specify the groups. If your experiment is well controlled and has worked well, what we hope to see is that the greatest sources of variation in the data are the treatments/groups we are interested in. It is also an incredibly useful tool for quality control and checking for outliers. We can use the <code>reduce_dimensions</code> function to calculate the dimensions.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get principal components</span>
<span class="va">counts_scal_MDS</span> <span class="op">&lt;-</span>
  <span class="va">counts_scaled</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/reduce_dimensions-methods.html">reduce_dimensions</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"MDS"</span>, scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; tidybulk says: to access the raw results do `attr(..., "internals")$MDS`</span></code></pre></div>
<p>This joins the result to the counts object.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Take a look</span>
<span class="va">counts_scal_MDS</span>
<span class="co">#&gt; # A tibble: 191,628 x 17</span>
<span class="co">#&gt;    EntrezGeneID Length sample    counts sample_name symbol FileName   SampleName</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#&gt;  1 497097         3634 MCL1-DG_…    438 MCL1-DG     Xkr4   MCL1.DG_B… MCL1.DG   </span>
<span class="co">#&gt;  2 497097         3634 MCL1-DH_…    300 MCL1-DH     Xkr4   MCL1.DH_B… MCL1.DH   </span>
<span class="co">#&gt;  3 497097         3634 MCL1-DI_…     65 MCL1-DI     Xkr4   MCL1.DI_B… MCL1.DI   </span>
<span class="co">#&gt;  4 497097         3634 MCL1-DJ_…    237 MCL1-DJ     Xkr4   MCL1.DJ_B… MCL1.DJ   </span>
<span class="co">#&gt;  5 497097         3634 MCL1-DK_…    354 MCL1-DK     Xkr4   MCL1.DK_B… MCL1.DK   </span>
<span class="co">#&gt;  6 497097         3634 MCL1-DL_…    287 MCL1-DL     Xkr4   MCL1.DL_B… MCL1.DL   </span>
<span class="co">#&gt;  7 497097         3634 MCL1-LA_…      0 MCL1-LA     Xkr4   MCL1.LA_B… MCL1.LA   </span>
<span class="co">#&gt;  8 497097         3634 MCL1-LB_…      0 MCL1-LB     Xkr4   MCL1.LB_B… MCL1.LB   </span>
<span class="co">#&gt;  9 497097         3634 MCL1-LC_…      0 MCL1-LC     Xkr4   MCL1.LC_B… MCL1.LC   </span>
<span class="co">#&gt; 10 497097         3634 MCL1-LD_…      0 MCL1-LD     Xkr4   MCL1.LD_B… MCL1.LD   </span>
<span class="co">#&gt; # … with 191,618 more rows, and 9 more variables: Group &lt;chr&gt;, CellType &lt;chr&gt;,</span>
<span class="co">#&gt; #   Status &lt;chr&gt;, .abundant &lt;lgl&gt;, TMM &lt;dbl&gt;, multiplier &lt;dbl&gt;,</span>
<span class="co">#&gt; #   counts_scaled &lt;dbl&gt;, Dim1 &lt;dbl&gt;, Dim2 &lt;dbl&gt;</span></code></pre></div>
<p>For plotting, we can select just the sample-wise information with <code>pivot_sample</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take a look</span>
<span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_sample-methods.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 x 12</span>
<span class="co">#&gt;    sample_name sample  FileName SampleName Group CellType Status .abundant   TMM</span>
<span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;  &lt;lgl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 MCL1-DG     MCL1-D… MCL1.DG… MCL1.DG    basa… basal    virgin TRUE      1.25 </span>
<span class="co">#&gt;  2 MCL1-DH     MCL1-D… MCL1.DH… MCL1.DH    basa… basal    virgin TRUE      1.22 </span>
<span class="co">#&gt;  3 MCL1-DI     MCL1-D… MCL1.DI… MCL1.DI    basa… basal    pregn… TRUE      1.14 </span>
<span class="co">#&gt;  4 MCL1-DJ     MCL1-D… MCL1.DJ… MCL1.DJ    basa… basal    pregn… TRUE      1.09 </span>
<span class="co">#&gt;  5 MCL1-DK     MCL1-D… MCL1.DK… MCL1.DK    basa… basal    lacta… TRUE      1.05 </span>
<span class="co">#&gt;  6 MCL1-DL     MCL1-D… MCL1.DL… MCL1.DL    basa… basal    lacta… TRUE      1.11 </span>
<span class="co">#&gt;  7 MCL1-LA     MCL1-L… MCL1.LA… MCL1.LA    lumi… luminal  virgin TRUE      1.38 </span>
<span class="co">#&gt;  8 MCL1-LB     MCL1-L… MCL1.LB… MCL1.LB    lumi… luminal  virgin TRUE      1.38 </span>
<span class="co">#&gt;  9 MCL1-LC     MCL1-L… MCL1.LC… MCL1.LC    lumi… luminal  pregn… TRUE      1.02 </span>
<span class="co">#&gt; 10 MCL1-LD     MCL1-L… MCL1.LD… MCL1.LD    lumi… luminal  pregn… TRUE      0.924</span>
<span class="co">#&gt; 11 MCL1-LE     MCL1-L… MCL1.LE… MCL1.LE    lumi… luminal  lacta… TRUE      0.500</span>
<span class="co">#&gt; 12 MCL1-LF     MCL1-L… MCL1.LF… MCL1.LF    lumi… luminal  lacta… TRUE      0.504</span>
<span class="co">#&gt; # … with 3 more variables: multiplier &lt;dbl&gt;, Dim1 &lt;dbl&gt;, Dim2 &lt;dbl&gt;</span></code></pre></div>
<p>We can now plot the reduced dimensions.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MDS plot</span>
<span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_sample-methods.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, colour <span class="op">=</span> <span class="va">CellType</span>, shape <span class="op">=</span> <span class="va">Status</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-23-1.png" width="70%"></p>
</div>
<div id="hierarchical-clustering-with-heatmaps" class="section level3">
<h3 class="hasAnchor">
<a href="#hierarchical-clustering-with-heatmaps" class="anchor"></a>Hierarchical clustering with heatmaps</h3>
<p>An alternative to principal component analysis for examining relationships between samples is using hierarchical clustering. Heatmaps are a nice visualisation to examine hierarchical clustering of your samples. tidybulk has a simple function we can use, <code>keep_variable</code>, to extract the most variable genes which we can then plot with tidyHeatmap.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>

  <span class="co"># extract 500 most variable genes</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/keep_variable-methods.html">keep_variable</a></span><span class="op">(</span>.abundance <span class="op">=</span> <span class="va">counts_scaled</span>, top <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># create heatmap</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/heatmap-methods.html">heatmap</a></span><span class="op">(</span>
    .column <span class="op">=</span> <span class="va">sample_name</span>,
    .row <span class="op">=</span> <span class="va">EntrezGeneID</span>,
    .value <span class="op">=</span> <span class="va">counts_scaled</span>,
    transform <span class="op">=</span> <span class="va">log1p</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_tile-methods.html">add_tile</a></span><span class="op">(</span><span class="va">CellType</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_tile-methods.html">add_tile</a></span><span class="op">(</span><span class="va">Status</span><span class="op">)</span>
<span class="co">#&gt; Getting the 500 most variable genes</span>
<span class="co">#&gt; tidyHeatmap says: (once per session) from release 1.2.3 the grouping labels have white background by default. To add color for one-ay grouping specify palette_grouping = list(c("red", "blue"))</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-24-1.png" width="70%"></p>
</div>
</div>
<div id="differential-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential expression</h2>
<p><em>tidybulk</em> integrates several popular methods for differential transcript abundance testing: the edgeR quasi-likelihood <span class="citation">[@chen2016reads]</span> (tidybulk default method), edgeR likelihood ratio <span class="citation">[@mccarthy2012differential]</span>, limma-voom <span class="citation">[@law2014voom]</span> and DESeq2 <span class="citation">[@love2014moderated]</span>. Here’s we’ll run limma-voom, the same method used in the original workshop that this material is based on.</p>
<p>We give <code>test_differential_abundance</code> our tidybulk counts object and a formula, specifying the column that contains our groups to be compared.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">&lt;-</span> <span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
                                 method <span class="op">=</span> <span class="st">"limma_voom"</span>,
                                .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
                                omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: All testing methods use raw counts, irrespective of if scale_abundance</span>
<span class="co">#&gt; or adjust_abundance have been calculated. Therefore, it is essential to add covariates</span>
<span class="co">#&gt; such as batch effects (if applicable) in the formula.</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: The design column names are "Groupbasal.lactate, Groupbasal.pregnant, Groupbasal.virgin, Groupluminal.lactate, Groupluminal.pregnant, Groupluminal.virgin"</span>
<span class="co">#&gt; tidybulk says: to access the raw results (fitted GLM) do `attr(..., "internals")$voom`</span></code></pre></div>
<p>This outputs the columns from each method such as log-fold change (logFC), false-discovery rate (adj.P.Val) and probability value (P.Value). logFC is log2(basal.pregnant/basal.lactate).</p>
<p>We can count the number of differentially expressed genes.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>signif<span class="op">=</span><span class="va">adj.P.Val</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Up<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">signif</span> <span class="op">&amp;</span> <span class="va">logFC</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span>, 
              Down<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">signif</span> <span class="op">&amp;</span> <span class="va">logFC</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span>,
              NotSig<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">!</span><span class="va">signif</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 3</span>
<span class="co">#&gt;      Up  Down NotSig</span>
<span class="co">#&gt;   &lt;int&gt; &lt;int&gt;  &lt;int&gt;</span>
<span class="co">#&gt; 1  2702  2638  10629</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de_two_contrasts</span> <span class="op">&lt;-</span> <span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
                                 method <span class="op">=</span> <span class="st">"limma_voom"</span>,
                                .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span>, <span class="st">"Groupluminal.pregnant - Groupluminal.lactate"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: All testing methods use raw counts, irrespective of if scale_abundance</span>
<span class="co">#&gt; or adjust_abundance have been calculated. Therefore, it is essential to add covariates</span>
<span class="co">#&gt; such as batch effects (if applicable) in the formula.</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: The design column names are "Groupbasal.lactate, Groupbasal.pregnant, Groupbasal.virgin, Groupluminal.lactate, Groupluminal.pregnant, Groupluminal.virgin"</span>
<span class="co">#&gt; tidybulk says: to access the raw results (fitted GLM) do `attr(..., "internals")$voom`</span></code></pre></div>
</div>
<div id="writing-out-the-results" class="section level2">
<h2 class="hasAnchor">
<a href="#writing-out-the-results" class="anchor"></a>Writing out the results</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span>
    <span class="va">pivot_transcript</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_tsv</a></span><span class="op">(</span><span class="st">"de_results.tsv"</span><span class="op">)</span></code></pre></div>
</div>
<div id="plots-after-testing-for-differentially-expressed" class="section level2">
<h2 class="hasAnchor">
<a href="#plots-after-testing-for-differentially-expressed" class="anchor"></a>Plots after testing for differentially expressed</h2>
<p>We’ll extract the symbols for a few top genes (by P value) to use in some of the plots we will make.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">topgenes_symbols</span> <span class="op">&lt;-</span>
  <span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_min</a></span><span class="op">(</span><span class="va">P.Value</span>, n<span class="op">=</span><span class="fl">6</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></code></pre></div>
<div id="ma-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#ma-plots" class="anchor"></a>MA plots</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="co"># Specify significant genes    </span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>signif<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span>, <span class="st">"Up"</span>,
                <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">logFC</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, <span class="st">"Down"</span>, 
                <span class="st">"NotSig"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Specify genes to label</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">%in%</span> <span class="va">topgenes_symbols</span>, <span class="va">symbol</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">AveExpr</span>, y <span class="op">=</span> <span class="va">logFC</span>, colour<span class="op">=</span><span class="va">signif</span>, label<span class="op">=</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Up"</span><span class="op">=</span><span class="st">"red"</span>, <span class="st">"Down"</span><span class="op">=</span><span class="st">"blue"</span>, <span class="st">"NotSig"</span><span class="op">=</span><span class="st">"grey"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span>colour<span class="op">=</span><span class="st">"black"</span>, max.overlaps <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Subset data</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>significant <span class="op">=</span> <span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>transcript <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span>, <span class="va">symbol</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Plot</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">AveExpr</span>, y <span class="op">=</span> <span class="va">logFC</span>, label <span class="op">=</span> <span class="va">transcript</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">significant</span>, size <span class="op">=</span> <span class="va">significant</span>, alpha <span class="op">=</span> <span class="va">significant</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"#e11f28"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_discrete</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Warning: Using size for a discrete variable is not advised.</span>
<span class="co">#&gt; Warning: Using alpha for a discrete variable is not advised.</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-31-1.png" width="700"></p>
</div>
<div id="volcano-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h3>
<p>Volcano plots are a useful genome-wide plot for checking that the analysis looks good. Volcano plots enable us to visualise the significance of change (p-value) versus the fold change (logFC). Highly significant genes are towards the top of the plot. We can also colour significant genes (e.g. genes with false-discovery rate &lt; 0.05)</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Subset data</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>significant <span class="op">=</span> <span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">%in%</span> <span class="va">topgenes_symbols</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># Plot</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="va">P.Value</span>, label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">significant</span>, size <span class="op">=</span> <span class="va">significant</span>, alpha <span class="op">=</span> <span class="va">significant</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>

  <span class="co"># Custom scales</span>
  <span class="va">custom_theme</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="st">"log10_reverse"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"#e11f28"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_discrete</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'colour' is already present. Adding another scale for 'colour',</span>
<span class="co">#&gt; which will replace the existing scale.</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-32-1.png" width="70%"></p>
</div>
<div id="stripcharts" class="section level3">
<h3 class="hasAnchor">
<a href="#stripcharts" class="anchor"></a>Stripcharts</h3>
<p>Before following up on the differentially expressed genes with further lab work, it is also recommended to have a look at the expression levels of the individual samples for the genes of interest. We can use stripcharts to do this. These will help show if expression is consistent amongst replicates in the groups.</p>
<p>With stripcharts we can see if replicates tend to group together and how the expression compares to the other groups. We’ll also add a box plot to show the distribution. Tidyverse faceting makes it easy to create a plot for each gene.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strip_chart</span> <span class="op">&lt;-</span>
  <span class="va">counts_scaled</span> <span class="op">%&gt;%</span>

  <span class="co"># extract counts for top differentially expressed genes</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">%in%</span> <span class="va">topgenes_symbols</span><span class="op">)</span> <span class="op">%&gt;%</span>

  <span class="co"># make faceted stripchart</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Group</span>, y <span class="op">=</span> <span class="va">counts_scaled</span> <span class="op">+</span> <span class="fl">1</span>, colour <span class="op">=</span> <span class="va">Group</span>, label <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">symbol</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span> <span class="op">+</span>
  <span class="co"># some plot customisations</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span>    
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span><span class="op">)</span>
    

<span class="va">strip_chart</span></code></pre></div>
<p><img src="tidytranscriptomics_files/figure-html/unnamed-chunk-33-1.png" width="70%"></p>
</div>
</div>
<div id="interactive-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#interactive-plots" class="anchor"></a>Interactive Plots</h2>
<p>A really nice feature of using tidyverse and ggplot2 is that we can make interactive plots quite easily using the plotly package. This can be very useful for exploring what genes or samples are in the plots. We can make interactive plots directly from our ggplot2 object (strip_chart). Having <code>label</code> in the <code>aes</code> is useful to visualise the identifier of the data point (here the sample id) or other variables when we hover over the plot.</p>
<p>We can also specify which parameters from the <code>aes</code> we want to show up when we hover over the plot with <code>tooltip</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strip_chart</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://docs.ropensci.org/plotly/reference/ggplotly.html">ggplotly</a></span><span class="op">(</span>tooltip <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"label"</span>, <span class="st">"y"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-c14a7681d4f63eae4c7c" style="width:70%;height:432.632880098888px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-c14a7681d4f63eae4c7c">{"x":{"data":[{"x":[1.21916934289038,1.13651031702757],"y":[3.84620256165716,3.76109156700645],"text":["counts_scaled + 1: 7017.015971<br />sample_name: MCL1-DK","counts_scaled + 1: 5765.864086<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,1)","legendgroup":"(basal.lactate,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.32645822670311,0.890119272656739],"y":[3.14087763113056,3.0543760255411],"text":["counts_scaled + 1: 1383.348388<br />sample_name: MCL1-DK","counts_scaled + 1: 1133.384017<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,2)","legendgroup":"(basal.lactate,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.716689457558095,1.13505574651062],"y":[3.06184156987147,3.08652609264852],"text":["counts_scaled + 1: 1153.754192<br />sample_name: MCL1-DK","counts_scaled + 1: 1220.641313<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,3)","legendgroup":"(basal.lactate,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.918576847948134,0.657776193507016],"y":[2.69981324553801,2.61980339295557],"text":["counts_scaled + 1:  501.324019<br />sample_name: MCL1-DK","counts_scaled + 1:  416.697682<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,4)","legendgroup":"(basal.lactate,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.934113693237305,0.691024477221072],"y":[3.43750405257966,3.36031174204075],"text":["counts_scaled + 1: 2736.997502<br />sample_name: MCL1-DK","counts_scaled + 1: 2293.219768<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,5)","legendgroup":"(basal.lactate,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.808568814024329,1.13079730048776],"y":[3.33166733981937,3.33296832530511],"text":["counts_scaled + 1: 2146.749090<br />sample_name: MCL1-DK","counts_scaled + 1: 2151.059003<br />sample_name: MCL1-DL"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(230,159,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(230,159,0,1)"}},"hoveron":"points","name":"(basal.lactate,6)","legendgroup":"(basal.lactate,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.98357848413289,2.08952214978635],"y":[3.11850768763824,3.19844739939862],"text":["counts_scaled + 1: 1313.252533<br />sample_name: MCL1-DI","counts_scaled + 1: 1579.729614<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,1)","legendgroup":"(basal.pregnant,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.2381503501907,2.21583432089537],"y":[3.97451247293095,3.91119462281458],"text":["counts_scaled + 1: 9434.585204<br />sample_name: MCL1-DI","counts_scaled + 1: 8146.189388<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,2)","legendgroup":"(basal.pregnant,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.29634615015239,2.07261788006872],"y":[2.12949004892998,2.2850793996626],"text":["counts_scaled + 1:  134.742764<br />sample_name: MCL1-DI","counts_scaled + 1:  192.734293<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,3)","legendgroup":"(basal.pregnant,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.91428099423647,1.72190637197345],"y":[3.35422167810661,3.32192965608592],"text":["counts_scaled + 1: 2260.465992<br />sample_name: MCL1-DI","counts_scaled + 1: 2099.523041<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,4)","legendgroup":"(basal.pregnant,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[2.2620493484661,1.91102710757405],"y":[2.64970225535584,2.74915284474943],"text":["counts_scaled + 1:  446.284732<br />sample_name: MCL1-DI","counts_scaled + 1:  561.251122<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,5)","legendgroup":"(basal.pregnant,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[2.34185067173094,1.71951166894287],"y":[3.87688886404258,3.88263487619808],"text":["counts_scaled + 1: 7532.291065<br />sample_name: MCL1-DI","counts_scaled + 1: 7637.829658<br />sample_name: MCL1-DJ"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(86,180,233,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(86,180,233,1)"}},"hoveron":"points","name":"(basal.pregnant,6)","legendgroup":"(basal.pregnant,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.17533544991165,3.03002963364124],"y":[3.0044499611418,2.98015963481416],"text":["counts_scaled + 1: 1010.583419<br />sample_name: MCL1-DG","counts_scaled + 1:  955.907285<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,1)","legendgroup":"(basal.virgin,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.19279438536614,2.8859814427793],"y":[3.40676786222464,3.53017046115652],"text":["counts_scaled + 1: 2550.347479<br />sample_name: MCL1-DG","counts_scaled + 1: 3387.618235<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,2)","legendgroup":"(basal.virgin,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.32082061432302,2.80183795448393],"y":[2.25759541918853,2.3133019823081],"text":["counts_scaled + 1:  180.962725<br />sample_name: MCL1-DG","counts_scaled + 1:  205.681237<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,3)","legendgroup":"(basal.virgin,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2.65632509905845,2.96173083651811],"y":[2.80200336140375,2.78574663983224],"text":["counts_scaled + 1:  633.483103<br />sample_name: MCL1-DG","counts_scaled + 1:  610.966395<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,4)","legendgroup":"(basal.virgin,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.15121430531144,2.75197108499706],"y":[3.09196048998692,3.07129340597226],"text":["counts_scaled + 1: 1235.350142<br />sample_name: MCL1-DG","counts_scaled + 1: 1179.344173<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,5)","legendgroup":"(basal.virgin,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.2399147067219,2.81436931360513],"y":[3.59841937617769,3.62735970691206],"text":["counts_scaled + 1: 3966.153812<br />sample_name: MCL1-DG","counts_scaled + 1: 4238.146243<br />sample_name: MCL1-DH"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,158,115,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,158,115,1)"}},"hoveron":"points","name":"(basal.virgin,6)","legendgroup":"(basal.virgin,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.23567169625312,3.86838176120073],"y":[1.55793400943751,1.48659916484737],"text":["counts_scaled + 1:   36.143844<br />sample_name: MCL1-LE","counts_scaled + 1:   30.678726<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,1)","legendgroup":"(luminal.lactate,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.3533463191241,4.38737310618162],"y":[5.98499298356464e-05,0.65238400325414],"text":["counts_scaled + 1:    1.000000<br />sample_name: MCL1-LE","counts_scaled + 1:    4.491615<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,2)","legendgroup":"(luminal.lactate,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.28164431694895,3.78437332250178],"y":[1.26874963695758,1.48651396165182],"text":["counts_scaled + 1:   18.571922<br />sample_name: MCL1-LE","counts_scaled + 1:   30.678726<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,3)","legendgroup":"(luminal.lactate,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.62504451032728,3.8533839520067],"y":[2.59443624856543,2.67399883646339],"text":["counts_scaled + 1:  392.853858<br />sample_name: MCL1-LE","counts_scaled + 1:  472.368003<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,4)","legendgroup":"(luminal.lactate,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.83716204054654,4.07381335590035],"y":[3.51944545513189,3.56915410718928],"text":["counts_scaled + 1: 3308.035695<br />sample_name: MCL1-LE","counts_scaled + 1: 3709.094960<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,5)","legendgroup":"(luminal.lactate,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[3.96801605131477,3.94516847301275],"y":[0.440488760555347,0.438792345477039],"text":["counts_scaled + 1:    2.757192<br />sample_name: MCL1-LE","counts_scaled + 1:    2.745807<br />sample_name: MCL1-LF"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(240,228,66,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(240,228,66,1)"}},"hoveron":"points","name":"(luminal.lactate,6)","legendgroup":"(luminal.lactate,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.39794265255332,4.84061452839524],"y":[1.50123593013364,1.54672370833841],"text":["counts_scaled + 1:   31.717333<br />sample_name: MCL1-LC","counts_scaled + 1:   35.187467<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,1)","legendgroup":"(luminal.pregnant,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[5.02943590320647,5.1825545353815],"y":[1.36352879859238,1.14064310927228],"text":["counts_scaled + 1:   23.078083<br />sample_name: MCL1-LC","counts_scaled + 1:   13.820300<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,2)","legendgroup":"(luminal.pregnant,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[5.21929564848542,4.98625228460878],"y":[1.09757565523087,1.14069175534072],"text":["counts_scaled + 1:   12.519000<br />sample_name: MCL1-LC","counts_scaled + 1:   13.820300<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,3)","legendgroup":"(luminal.pregnant,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.84956642687321,5.06650664955378],"y":[2.84025449680011,2.92107936619767],"text":["counts_scaled + 1:  692.139985<br />sample_name: MCL1-LC","counts_scaled + 1:  834.319508<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,4)","legendgroup":"(luminal.pregnant,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.76257485244423,5.02770874500275],"y":[3.24765700703377,3.23645621576273],"text":["counts_scaled + 1: 1768.206544<br />sample_name: MCL1-LC","counts_scaled + 1: 1723.193649<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,5)","legendgroup":"(luminal.pregnant,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[4.9200486574322,4.62619201987982],"y":[1.23851674575681,1.38913589850762],"text":["counts_scaled + 1:   17.318583<br />sample_name: MCL1-LC","counts_scaled + 1:   24.503884<br />sample_name: MCL1-LD"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,178,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,114,178,1)"}},"hoveron":"points","name":"(luminal.pregnant,6)","legendgroup":"(luminal.pregnant,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.87786063123494,6.08656726833433],"y":[2.14960741616823,2.15449090310562],"text":["counts_scaled + 1:  141.072623<br />sample_name: MCL1-LA","counts_scaled + 1:  142.810423<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,1)","legendgroup":"(luminal.virgin,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6.00140486396849,6.25835777502507],"y":[1.52508459203636,1.42057841038427],"text":["counts_scaled + 1:   33.503040<br />sample_name: MCL1-LA","counts_scaled + 1:   26.323290<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,2)","legendgroup":"(luminal.virgin,2)","showlegend":true,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6.004057408683,6.09277494437993],"y":[1.48307927913479,1.63339806367747],"text":["counts_scaled + 1:   30.407512<br />sample_name: MCL1-LA","counts_scaled + 1:   42.964309<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,3)","legendgroup":"(luminal.virgin,3)","showlegend":true,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6.03986256383359,5.61581655908376],"y":[2.34246145035943,2.2830365519277],"text":["counts_scaled + 1:  220.008577<br />sample_name: MCL1-LA","counts_scaled + 1:  192.009958<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,4)","legendgroup":"(luminal.virgin,4)","showlegend":true,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.77421821448952,6.04020374417305],"y":[3.31213512429117,3.27656302127577],"text":["counts_scaled + 1: 2051.013145<br />sample_name: MCL1-LA","counts_scaled + 1: 1890.117427<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,5)","legendgroup":"(luminal.virgin,5)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[5.79347734637558,6.06965877898037],"y":[0.751827461361439,1.01733686244588],"text":["counts_scaled + 1:    5.643291<br />sample_name: MCL1-LA","counts_scaled + 1:   10.405793<br />sample_name: MCL1-LB"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(213,94,0,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(213,94,0,1)"}},"hoveron":"points","name":"(luminal.virgin,6)","legendgroup":"(luminal.virgin,6)","showlegend":true,"xaxis":"x3","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":40.5768157463462,"r":7.30593607305936,"b":117.495039075528,"l":58.7795765877958},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xaxis":{"domain":[0,0.32941943900848],"automargin":true,"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-90,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"yaxis":{"domain":[0.517391429538076,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.19866278122022,4.173235104081],"tickmode":"array","ticktext":["1","10","100","1000","10000"],"tickvals":[0,1,2,3,4],"categoryorder":"array","categoryarray":["1","10","100","1000","10000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"counts_scaled + 1","x":-0.0233054616616261,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":15.9402241594022},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"Atp2b4","x":0.16470971950424,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Ecrg4","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Igfbp6","x":0.83529028049576,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Micall2","x":0.16470971950424,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Myof","x":0.5,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Wif1","x":0.83529028049576,"y":0.482608570461924,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":12.7521793275218},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0.517391429538076,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":24.4416770444168,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.32941943900848,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.337247227658187,"x1":0.662752772341813,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":0.482608570461924},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.67058056099152,"x1":1,"y0":0,"y1":24.4416770444168,"yanchor":0.482608570461924,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-90,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0.337247227658187,0.662752772341813],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"xaxis3":{"type":"linear","autorange":false,"range":[0.4,6.6],"tickmode":"array","ticktext":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"tickvals":[1,2,3,4,5,6],"categoryorder":"array","categoryarray":["basal.lactate","basal.pregnant","basal.virgin","luminal.lactate","luminal.pregnant","luminal.virgin"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-90,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0.67058056099152,1],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"yaxis2":{"type":"linear","autorange":false,"range":[-0.19866278122022,4.173235104081],"tickmode":"array","ticktext":["1","10","100","1000","10000"],"tickvals":[0,1,2,3,4],"categoryorder":"array","categoryarray":["1","10","100","1000","10000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":12.7521793275218},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":true,"domain":[0,0.482608570461924],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":12.7521793275218}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"d3d9735fae6e":{"x":{},"y":{},"colour":{},"label":{},"type":"scatter"}},"cur_data":"d3d9735fae6e","visdat":{"d3d9735fae6e":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="testing-relative-to-a-threshold-treat" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-relative-to-a-threshold-treat" class="anchor"></a>Testing relative to a threshold (TREAT)</h2>
<p>When there is a lot of differential expression, sometimes we may want to cut-off on a fold change threshold as well as a p-value threshold so that we follow up on the most biologically significant genes. However, it is not recommended to simply rank by p-value and then discard genes with small logFC’s, as this has been shown to increase the false discovery rate. In other words, you are not controlling the false discovery rate at 5% any more. There is a function called treat that performs this style of analysis correctly <span class="citation">[@McCarthy and Smyth 2009]</span>. We simply need to specify a logFC cut-off with <code>test_differential_abundance</code> using <code>test_above_log2_fold_change=</code> and tidybulk will use treat to calculate the p-values taking into account the information about logFC. We’ll use a log2 fold change of 1 which is equivalent to a fold change of 2.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de_treat</span> <span class="op">&lt;-</span> <span class="va">counts_scal_MDS</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>,
                                 method <span class="op">=</span> <span class="st">"limma_voom"</span>,
                                .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>,
                                omit_contrast_in_colnames <span class="op">=</span> <span class="cn">TRUE</span>,
                                test_above_log2_fold_change <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: All testing methods use raw counts, irrespective of if scale_abundance</span>
<span class="co">#&gt; or adjust_abundance have been calculated. Therefore, it is essential to add covariates</span>
<span class="co">#&gt; such as batch effects (if applicable) in the formula.</span>
<span class="co">#&gt; =====================================</span>
<span class="co">#&gt; tidybulk says: The design column names are "Groupbasal.lactate, Groupbasal.pregnant, Groupbasal.virgin, Groupluminal.lactate, Groupluminal.pregnant, Groupluminal.virgin"</span>
<span class="co">#&gt; tidybulk says: to access the raw results (fitted GLM) do `attr(..., "internals")$voom`</span></code></pre></div>
<p>If we count the number of differentially expressed genes we can see there are less significant genes now.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_de_treat</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>signif<span class="op">=</span><span class="va">adj.P.Val</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Up<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">signif</span> <span class="op">&amp;</span> <span class="va">logFC</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span>, 
              Down<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">signif</span> <span class="op">&amp;</span> <span class="va">logFC</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span>,
              NotSig<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">!</span><span class="va">signif</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 3</span>
<span class="co">#&gt;      Up  Down NotSig</span>
<span class="co">#&gt;   &lt;int&gt; &lt;int&gt;  &lt;int&gt;</span>
<span class="co">#&gt; 1    74    54  15841</span></code></pre></div>
</div>
<div id="gene-set-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-testing" class="anchor"></a>Gene Set Testing</h2>
<p>Sometimes there is quite a long list of differentially expressed genes to interpret after a differential expression analysis, and it is usually infeasible to go through the list one gene at a time trying to understand it’s biological function. A common downstream procedure is gene set testing, which aims to understand which pathways/gene networks the differentially expressed genes are implicated in.</p>
<p>With tidybulk we can use <code>test_gene_enrichment</code> to perform gene set testing via EGSEA. EGSEA provides a choice of methods that can be used either individually or in combination to generate an ensemble score. Here we will demonstrate CAMERA, which is a good option for testing a very large number of gene sets, such as the MSigDB sets, as it is very fast. CAMERA is known as a competitive gene set test, however it has the advantage that it can take into account inter-gene correlation within each gene set (Wu and Smyth 2012).</p>
<p>EGSEA also provide collections of gene sets from the Broad Institute’s Molecular Signatures Database (MSigDB) for human, mouse and rat. The c2 gene sets contain &gt;5000 curated gene sets collected from a variety of places: BioCarta, KEGG, Pathway Interaction Database, Reactome as well as some published studies.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_egsea</span> <span class="op">&lt;-</span> <span class="va">counts_scaled</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_gene_enrichment-methods.html">test_gene_enrichment</a></span><span class="op">(</span>.entrez<span class="op">=</span><span class="va">EntrezGeneID</span>, 
                         .abundance<span class="op">=</span><span class="va">counts_scaled</span>, 
                         .formula<span class="op">=</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Group</span>, 
                         .contrasts <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Groupbasal.pregnant - Groupbasal.lactate"</span><span class="op">)</span>, 
                         species<span class="op">=</span><span class="st">"mouse"</span>, 
                         methods<span class="op">=</span><span class="st">"camera"</span>,
                         gene_sets<span class="op">=</span><span class="st">"c2"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Loading MSigDB Gene Sets ... "</span>
<span class="co">#&gt; [1] "Loaded gene sets for the collection c2 ..."</span>
<span class="co">#&gt; [1] "Indexed the collection c2 ..."</span>
<span class="co">#&gt; [1] "Created annotation for the collection c2 ..."</span>
<span class="co">#&gt; ##------ Wed Jul 14 17:42:01 2021 ------##</span>
<span class="co">#&gt; ##------ Wed Jul 14 17:42:04 2021 ------##</span>
<span class="co">#&gt; ##------ Wed Jul 14 17:42:04 2021 ------##</span>
<span class="co">#&gt; ##------ Wed Jul 14 17:42:19 2021 ------##</span></code></pre></div>
<p>The output is a tibble containing each gene set tested with P values and other information. A HTML report is also produced containing summary plots.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts_egsea</span>
<span class="co">#&gt; # A tibble: 4,725 x 11</span>
<span class="co">#&gt;    data_base pathway     web_page    p.value NGenes Direction    p.adj avg.logfc</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 c2        REACTOME_3… https://w… 3.36e-25    128 Up        1.21e-21     0.674</span>
<span class="co">#&gt;  2 c2        REACTOME_P… https://w… 5.11e-25    102 Up        1.21e-21     0.688</span>
<span class="co">#&gt;  3 c2        KEGG_RIBOS… https://w… 1.05e-24    102 Up        1.65e-21     0.695</span>
<span class="co">#&gt;  4 c2        REACTOME_T… https://w… 4.43e-23    176 Up        5.23e-20     0.658</span>
<span class="co">#&gt;  5 c2        REACTOME_S… https://w… 1.68e-22    130 Up        1.59e-19     0.679</span>
<span class="co">#&gt;  6 c2        REACTOME_N… https://w… 2.37e-21    132 Up        1.87e-18     0.686</span>
<span class="co">#&gt;  7 c2        REACTOME_I… https://w… 3.50e-21    122 Up        2.36e-18     0.693</span>
<span class="co">#&gt;  8 c2        REACTOME_F… https://w… 5.17e-17     56 Up        3.05e-14     0.667</span>
<span class="co">#&gt;  9 c2        REACTOME_I… https://w… 8.85e-17    169 Up        4.65e-14     0.691</span>
<span class="co">#&gt; 10 c2        REACTOME_A… https://w… 1.71e-16     67 Up        8.10e-14     0.654</span>
<span class="co">#&gt; # … with 4,715 more rows, and 3 more variables: avg.logfc.dir &lt;dbl&gt;,</span>
<span class="co">#&gt; #   direction &lt;dbl&gt;, significance &lt;dbl&gt;</span></code></pre></div>
</div>
<div id="automatic-bibliography" class="section level2">
<h2 class="hasAnchor">
<a href="#automatic-bibliography" class="anchor"></a>Automatic bibliography</h2>
<p>Tidybulk provides a handy function called <code>get_bibliography</code> that keeps track of the references for the methods used in your tidybulk workflow. The references are in BibTeX format and can be imported into your reference manager.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/get_bibliography-methods.html">get_bibliography</a></span><span class="op">(</span><span class="va">counts_egsea</span><span class="op">)</span>
<span class="co">#&gt;  @Article{tidybulk,</span>
<span class="co">#&gt;   title = {tidybulk: an R tidy framework for modular transcriptomic data analysis},</span>
<span class="co">#&gt;   author = {Stefano Mangiola and Ramyar Molania and Ruining Dong and Maria A. Doyle &amp; Anthony T. Papenfuss},</span>
<span class="co">#&gt;   journal = {Genome Biology},</span>
<span class="co">#&gt;   year = {2021},</span>
<span class="co">#&gt;   volume = {22},</span>
<span class="co">#&gt;   number = {42},</span>
<span class="co">#&gt;   url = {https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02233-7},</span>
<span class="co">#&gt;   }</span>
<span class="co">#&gt; @article{wickham2019welcome,</span>
<span class="co">#&gt;   title={Welcome to the Tidyverse},</span>
<span class="co">#&gt;   author={Wickham, Hadley and Averick, Mara and Bryan, Jennifer and Chang, Winston and McGowan, Lucy D'Agostino and Francois, Romain and Grolemund, Garrett and Hayes, Alex and Henry, Lionel and Hester, Jim and others},</span>
<span class="co">#&gt;   journal={Journal of Open Source Software},</span>
<span class="co">#&gt;   volume={4},</span>
<span class="co">#&gt;   number={43},</span>
<span class="co">#&gt;   pages={1686},</span>
<span class="co">#&gt;   year={2019}</span>
<span class="co">#&gt;  }</span>
<span class="co">#&gt; @article{alhamdoosh2017combining,</span>
<span class="co">#&gt;   title={Combining multiple tools outperforms individual methods in gene set enrichment analyses},</span>
<span class="co">#&gt;   author={Alhamdoosh, Monther and Ng, Milica and Wilson, Nicholas J and Sheridan, Julie M and Huynh, Huy and Wilson, Michael J and Ritchie, Matthew E},</span>
<span class="co">#&gt;   journal={Bioinformatics},</span>
<span class="co">#&gt;   volume={33},</span>
<span class="co">#&gt;   number={3},</span>
<span class="co">#&gt;   pages={414--424},</span>
<span class="co">#&gt;   year={2017},</span>
<span class="co">#&gt;   publisher={Oxford University Press}</span>
<span class="co">#&gt;   }</span>
<span class="co">#&gt; @article{liberzon2011molecular,</span>
<span class="co">#&gt;   title={Molecular signatures database (MSigDB) 3.0},</span>
<span class="co">#&gt;   author={Liberzon, Arthur and Subramanian, Aravind and Pinchback, Reid and Thorvaldsdottir, Helga and Tamayo, Pablo and Mesirov, Jill P},</span>
<span class="co">#&gt;   journal={Bioinformatics},</span>
<span class="co">#&gt;   volume={27},</span>
<span class="co">#&gt;   number={12},</span>
<span class="co">#&gt;   pages={1739--1740},</span>
<span class="co">#&gt;   year={2011},</span>
<span class="co">#&gt;   publisher={Oxford University Press}</span>
<span class="co">#&gt;  }</span>
<span class="co">#&gt; @article{Wu2012,</span>
<span class="co">#&gt;   doi = {10.1093/nar/gks461},</span>
<span class="co">#&gt;   url = {https://doi.org/10.1093/nar/gks461},</span>
<span class="co">#&gt;   year = {2012},</span>
<span class="co">#&gt;   month = may,</span>
<span class="co">#&gt;   publisher = {Oxford University Press ({OUP})},</span>
<span class="co">#&gt;   volume = {40},</span>
<span class="co">#&gt;   number = {17},</span>
<span class="co">#&gt;   pages = {e133--e133},</span>
<span class="co">#&gt;   author = {Di Wu and Gordon K. Smyth},</span>
<span class="co">#&gt;   title = {Camera: a competitive gene set test accounting for inter-gene correlation},</span>
<span class="co">#&gt;   journal = {Nucleic Acids Research}</span>
<span class="co">#&gt; }</span></code></pre></div>
</div>
</div>
<div id="contributing" class="section level1">
<h1 class="hasAnchor">
<a href="#contributing" class="anchor"></a>Contributing</h1>
<p>If you want to suggest improvements for this material or ask questions, you can do so as described <a href="https://github.com/mblue9/RNAseq-R-tidyverse/blob/master/CONTRIBUTING.md">here</a>.</p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>Record package and version information with <code>sessionInfo</code></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] tidybulk_1.4.0       tidyHeatmap_1.3.1    EGSEA_1.20.0        </span>
<span class="co">#&gt;  [4] pathview_1.32.0      topGO_2.44.0         SparseM_1.81        </span>
<span class="co">#&gt;  [7] GO.db_3.13.0         graph_1.70.0         AnnotationDbi_1.54.1</span>
<span class="co">#&gt; [10] IRanges_2.26.0       S4Vectors_0.30.0     gage_2.42.0         </span>
<span class="co">#&gt; [13] Biobase_2.52.0       BiocGenerics_0.38.0  ggrepel_0.9.1       </span>
<span class="co">#&gt; [16] plotly_4.9.4.1       ggplot2_3.3.5        stringr_1.4.0       </span>
<span class="co">#&gt; [19] readr_1.4.0          tidyr_1.1.3          dplyr_1.0.7         </span>
<span class="co">#&gt; [22] tibble_3.1.2        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] utf8_1.2.1                  tidyselect_1.1.1           </span>
<span class="co">#&gt;   [3] RSQLite_2.2.7               htmlwidgets_1.5.3          </span>
<span class="co">#&gt;   [5] grid_4.1.0                  BiocParallel_1.26.1        </span>
<span class="co">#&gt;   [7] R2HTML_2.3.2                munsell_0.5.0              </span>
<span class="co">#&gt;   [9] ScaledMatrix_1.0.0          preprocessCore_1.54.0      </span>
<span class="co">#&gt;  [11] codetools_0.2-18            mutoss_0.1-12              </span>
<span class="co">#&gt;  [13] ragg_1.1.3                  DT_0.18                    </span>
<span class="co">#&gt;  [15] KEGGdzPathwaysGEO_1.30.0    withr_2.4.2                </span>
<span class="co">#&gt;  [17] colorspace_2.0-2            highr_0.9                  </span>
<span class="co">#&gt;  [19] knitr_1.33                  SingleCellExperiment_1.14.1</span>
<span class="co">#&gt;  [21] labeling_0.4.2              MatrixGenerics_1.4.0       </span>
<span class="co">#&gt;  [23] Rdpack_2.1.2                KEGGgraph_1.52.0           </span>
<span class="co">#&gt;  [25] org.Rn.eg.db_3.13.0         GenomeInfoDbData_1.2.6     </span>
<span class="co">#&gt;  [27] mnormt_2.0.2                hwriter_1.3.2              </span>
<span class="co">#&gt;  [29] farver_2.1.0                pheatmap_1.0.12            </span>
<span class="co">#&gt;  [31] bit64_4.0.5                 rhdf5_2.36.0               </span>
<span class="co">#&gt;  [33] rprojroot_2.0.2             vctrs_0.3.8                </span>
<span class="co">#&gt;  [35] generics_0.1.0              TH.data_1.0-10             </span>
<span class="co">#&gt;  [37] xfun_0.24                   dittoSeq_1.4.1             </span>
<span class="co">#&gt;  [39] R6_2.5.0                    doParallel_1.0.16          </span>
<span class="co">#&gt;  [41] GenomeInfoDb_1.28.1         clue_0.3-59                </span>
<span class="co">#&gt;  [43] Glimma_2.2.0                rsvd_1.0.5                 </span>
<span class="co">#&gt;  [45] locfit_1.5-9.4              bitops_1.0-7               </span>
<span class="co">#&gt;  [47] rhdf5filters_1.4.0          cachem_1.0.5               </span>
<span class="co">#&gt;  [49] DelayedArray_0.18.0         scales_1.1.1               </span>
<span class="co">#&gt;  [51] multcomp_1.4-17             gtable_0.3.0               </span>
<span class="co">#&gt;  [53] beachmat_2.8.0              org.Mm.eg.db_3.13.0        </span>
<span class="co">#&gt;  [55] Cairo_1.5-12.2              sandwich_3.0-1             </span>
<span class="co">#&gt;  [57] rlang_0.4.11                genefilter_1.74.0          </span>
<span class="co">#&gt;  [59] systemfonts_1.0.2           GlobalOptions_0.1.2        </span>
<span class="co">#&gt;  [61] splines_4.1.0               lazyeval_0.2.2             </span>
<span class="co">#&gt;  [63] PADOG_1.34.0                yaml_2.2.1                 </span>
<span class="co">#&gt;  [65] crosstalk_1.1.1             tools_4.1.0                </span>
<span class="co">#&gt;  [67] ellipsis_0.3.2              gplots_3.1.1               </span>
<span class="co">#&gt;  [69] RColorBrewer_1.1-2          HTMLUtils_0.1.7            </span>
<span class="co">#&gt;  [71] ggridges_0.5.3              plyr_1.8.6                 </span>
<span class="co">#&gt;  [73] TFisher_0.2.0               Rcpp_1.0.7                 </span>
<span class="co">#&gt;  [75] sparseMatrixStats_1.4.0     zlibbioc_1.38.0            </span>
<span class="co">#&gt;  [77] purrr_0.3.4                 RCurl_1.98-1.3             </span>
<span class="co">#&gt;  [79] viridis_0.6.1               GetoptLong_1.0.5           </span>
<span class="co">#&gt;  [81] cowplot_1.1.1               hgu133plus2.db_3.13.0      </span>
<span class="co">#&gt;  [83] zoo_1.8-9                   cluster_2.1.2              </span>
<span class="co">#&gt;  [85] SummarizedExperiment_1.22.0 fs_1.5.0                   </span>
<span class="co">#&gt;  [87] magrittr_2.0.1              data.table_1.14.0          </span>
<span class="co">#&gt;  [89] circlize_0.4.13             tmvnsim_1.0-2              </span>
<span class="co">#&gt;  [91] mvtnorm_1.1-2               matrixStats_0.59.0         </span>
<span class="co">#&gt;  [93] hms_1.1.0                   evaluate_0.14              </span>
<span class="co">#&gt;  [95] GSVA_1.40.1                 xtable_1.8-4               </span>
<span class="co">#&gt;  [97] globaltest_5.46.0           XML_3.99-0.6               </span>
<span class="co">#&gt;  [99] hgu133a.db_3.13.0           gridExtra_2.3              </span>
<span class="co">#&gt; [101] shape_1.4.6                 EGSEAdata_1.20.0           </span>
<span class="co">#&gt; [103] compiler_4.1.0              safe_3.32.0                </span>
<span class="co">#&gt; [105] KernSmooth_2.23-20          crayon_1.4.1               </span>
<span class="co">#&gt; [107] htmltools_0.5.1.1           geneplotter_1.70.0         </span>
<span class="co">#&gt; [109] DBI_1.1.1                   ComplexHeatmap_2.8.0       </span>
<span class="co">#&gt; [111] MASS_7.3-54                 Matrix_1.3-4               </span>
<span class="co">#&gt; [113] cli_3.0.0                   rbibutils_2.2.1            </span>
<span class="co">#&gt; [115] metap_1.4                   GenomicRanges_1.44.0       </span>
<span class="co">#&gt; [117] pkgconfig_2.0.3             pkgdown_1.6.1              </span>
<span class="co">#&gt; [119] sn_2.0.0                    numDeriv_2016.8-1.1        </span>
<span class="co">#&gt; [121] foreach_1.5.1               annotate_1.70.0            </span>
<span class="co">#&gt; [123] rngtools_1.5                multtest_2.48.0            </span>
<span class="co">#&gt; [125] XVector_0.32.0              doRNG_1.8.2                </span>
<span class="co">#&gt; [127] digest_0.6.27               Biostrings_2.60.1          </span>
<span class="co">#&gt; [129] rmarkdown_2.9               dendextend_1.15.1          </span>
<span class="co">#&gt; [131] edgeR_3.34.0                DelayedMatrixStats_1.14.0  </span>
<span class="co">#&gt; [133] GSEABase_1.54.0             curl_4.3.2                 </span>
<span class="co">#&gt; [135] gtools_3.9.2                rjson_0.2.20               </span>
<span class="co">#&gt; [137] GSA_1.03.1                  lifecycle_1.0.0            </span>
<span class="co">#&gt; [139] nlme_3.1-152                jsonlite_1.7.2             </span>
<span class="co">#&gt; [141] Rhdf5lib_1.14.2             desc_1.3.0                 </span>
<span class="co">#&gt; [143] viridisLite_0.4.0           limma_3.48.1               </span>
<span class="co">#&gt; [145] fansi_0.5.0                 pillar_1.6.1               </span>
<span class="co">#&gt; [147] lattice_0.20-44             KEGGREST_1.32.0            </span>
<span class="co">#&gt; [149] fastmap_1.1.0               httr_1.4.2                 </span>
<span class="co">#&gt; [151] plotrix_3.8-1               survival_3.2-11            </span>
<span class="co">#&gt; [153] glue_1.4.2                  png_0.1-7                  </span>
<span class="co">#&gt; [155] iterators_1.0.13            bit_4.0.4                  </span>
<span class="co">#&gt; [157] Rgraphviz_2.36.0            stringi_1.7.2              </span>
<span class="co">#&gt; [159] HDF5Array_1.20.0            blob_1.2.1                 </span>
<span class="co">#&gt; [161] textshaping_0.3.5           DESeq2_1.32.0              </span>
<span class="co">#&gt; [163] org.Hs.eg.db_3.13.0         BiocSingular_1.8.1         </span>
<span class="co">#&gt; [165] caTools_1.18.2              memoise_2.0.0              </span>
<span class="co">#&gt; [167] mathjaxr_1.4-0              irlba_2.3.3</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>&lt;maria.doyle at petermac.org&gt;<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>&lt;mangiola.s at wehi.edu.au&gt;<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Doyle, Stefano Mangiola.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
